// @apideck/file-picker/dist/file-picker.esm.js
import React, { useState, useEffect, Fragment, useCallback, useContext, createContext, useRef, useLayoutEffect, forwardRef } from 'react';
import { Transition, Menu } from '@headlessui/react';
import { createPortal } from 'react-dom';
import { useTable, useSortBy } from 'react-table';
import { Waypoint } from 'react-waypoint';
import { useDropzone } from 'react-dropzone';
import useSWR from 'swr';

function asyncGeneratorStep(gen, resolve, reject, _next, _throw, key, arg) {
  try {
    var info = gen[key](arg);
    var value = info.value;
  } catch (error) {
    reject(error);
    return;
  }

  if (info.done) {
    resolve(value);
  } else {
    Promise.resolve(value).then(_next, _throw);
  }
}

function _asyncToGenerator(fn) {
  return function () {
    var self = this,
        args = arguments;
    return new Promise(function (resolve, reject) {
      var gen = fn.apply(self, args);

      function _next(value) {
        asyncGeneratorStep(gen, resolve, reject, _next, _throw, "next", value);
      }

      function _throw(err) {
        asyncGeneratorStep(gen, resolve, reject, _next, _throw, "throw", err);
      }

      _next(undefined);
    });
  };
}

function _extends() {
  _extends = Object.assign || function (target) {
    for (var i = 1; i < arguments.length; i++) {
      var source = arguments[i];

      for (var key in source) {
        if (Object.prototype.hasOwnProperty.call(source, key)) {
          target[key] = source[key];
        }
      }
    }

    return target;
  };

  return _extends.apply(this, arguments);
}

function _objectWithoutPropertiesLoose(source, excluded) {
  if (source == null) return {};
  var target = {};
  var sourceKeys = Object.keys(source);
  var key, i;

  for (i = 0; i < sourceKeys.length; i++) {
    key = sourceKeys[i];
    if (excluded.indexOf(key) >= 0) continue;
    target[key] = source[key];
  }

  return target;
}

var Logo = function Logo(_ref) {
  var _ref$className = _ref.className,
      className = _ref$className === void 0 ? '' : _ref$className;
  return React.createElement("svg", {
    width: 95,
    height: 23,
    viewBox: "0 0 95 23",
    fill: "none",
    xmlns: "http://www.w3.org/2000/svg",
    className: className
  }, React.createElement("path", {
    d: "M1.2 15.05C1.25 14.2167 1.45833 13.525 1.825 12.975C2.19167 12.425 2.65833 11.9833 3.225 11.65C3.79167 11.3167 4.425 11.0833 5.125 10.95C5.84167 10.8 6.55833 10.725 7.275 10.725C7.925 10.725 8.58333 10.775 9.25 10.875C9.91667 10.9583 10.525 11.1333 11.075 11.4C11.625 11.6667 12.075 12.0417 12.425 12.525C12.775 12.9917 12.95 13.6167 12.95 14.4V21.125C12.95 21.7083 12.9833 22.2667 13.05 22.8C13.1167 23.3333 13.2333 23.7333 13.4 24H9.8C9.73333 23.8 9.675 23.6 9.625 23.4C9.59167 23.1833 9.56667 22.9667 9.55 22.75C8.98333 23.3333 8.31667 23.7417 7.55 23.975C6.78333 24.2083 6 24.325 5.2 24.325C4.58333 24.325 4.00833 24.25 3.475 24.1C2.94167 23.95 2.475 23.7167 2.075 23.4C1.675 23.0833 1.35833 22.6833 1.125 22.2C0.908333 21.7167 0.8 21.1417 0.8 20.475C0.8 19.7417 0.925 19.1417 1.175 18.675C1.44167 18.1917 1.775 17.8083 2.175 17.525C2.59167 17.2417 3.05833 17.0333 3.575 16.9C4.10833 16.75 4.64167 16.6333 5.175 16.55C5.70833 16.4667 6.23333 16.4 6.75 16.35C7.26667 16.3 7.725 16.225 8.125 16.125C8.525 16.025 8.84167 15.8833 9.075 15.7C9.30833 15.5 9.41667 15.2167 9.4 14.85C9.4 14.4667 9.33333 14.1667 9.2 13.95C9.08333 13.7167 8.91667 13.5417 8.7 13.425C8.5 13.2917 8.25833 13.2083 7.975 13.175C7.70833 13.125 7.41667 13.1 7.1 13.1C6.4 13.1 5.85 13.25 5.45 13.55C5.05 13.85 4.81667 14.35 4.75 15.05H1.2ZM9.4 17.675C9.25 17.8083 9.05833 17.9167 8.825 18C8.60833 18.0667 8.36667 18.125 8.1 18.175C7.85 18.225 7.58333 18.2667 7.3 18.3C7.01667 18.3333 6.73333 18.375 6.45 18.425C6.18333 18.475 5.91667 18.5417 5.65 18.625C5.4 18.7083 5.175 18.825 4.975 18.975C4.79167 19.1083 4.64167 19.2833 4.525 19.5C4.40833 19.7167 4.35 19.9917 4.35 20.325C4.35 20.6417 4.40833 20.9083 4.525 21.125C4.64167 21.3417 4.8 21.5167 5 21.65C5.2 21.7667 5.43333 21.85 5.7 21.9C5.96667 21.95 6.24167 21.975 6.525 21.975C7.225 21.975 7.76667 21.8583 8.15 21.625C8.53333 21.3917 8.81667 21.1167 9 20.8C9.18333 20.4667 9.29167 20.1333 9.325 19.8C9.375 19.4667 9.4 19.2 9.4 19V17.675ZM22.2305 21.675C21.6638 21.675 21.1805 21.5583 20.7805 21.325C20.3805 21.0917 20.0555 20.7917 19.8055 20.425C19.5721 20.0417 19.3971 19.6 19.2805 19.1C19.1805 18.6 19.1305 18.0917 19.1305 17.575C19.1305 17.0417 19.1805 16.525 19.2805 16.025C19.3805 15.525 19.5471 15.0833 19.7805 14.7C20.0305 14.3167 20.3471 14.0083 20.7305 13.775C21.1305 13.525 21.6221 13.4 22.2055 13.4C22.7721 13.4 23.2471 13.525 23.6305 13.775C24.0305 14.0083 24.3555 14.325 24.6055 14.725C24.8555 15.1083 25.0305 15.55 25.1305 16.05C25.2471 16.55 25.3055 17.0583 25.3055 17.575C25.3055 18.0917 25.2555 18.6 25.1555 19.1C25.0555 19.6 24.8805 20.0417 24.6305 20.425C24.3971 20.7917 24.0805 21.0917 23.6805 21.325C23.2971 21.5583 22.8138 21.675 22.2305 21.675ZM15.7055 11.075V28.525H19.2555V22.4H19.3055C19.7388 23.0333 20.2888 23.5167 20.9555 23.85C21.6388 24.1667 22.3805 24.325 23.1805 24.325C24.1305 24.325 24.9555 24.1417 25.6555 23.775C26.3721 23.4083 26.9638 22.9167 27.4305 22.3C27.9138 21.6833 28.2721 20.975 28.5055 20.175C28.7388 19.375 28.8555 18.5417 28.8555 17.675C28.8555 16.7583 28.7388 15.8833 28.5055 15.05C28.2721 14.2 27.9138 13.4583 27.4305 12.825C26.9471 12.1917 26.3388 11.6833 25.6055 11.3C24.8721 10.9167 23.9971 10.725 22.9805 10.725C22.1805 10.725 21.4471 10.8833 20.7805 11.2C20.1138 11.5167 19.5638 12.025 19.1305 12.725H19.0805V11.075H15.7055ZM34.6387 9.075V6.15H31.0887V9.075H34.6387ZM31.0887 11.075V24H34.6387V11.075H31.0887ZM46.609 17.5C46.609 18.0333 46.559 18.55 46.459 19.05C46.359 19.55 46.1923 20 45.959 20.4C45.7257 20.7833 45.409 21.0917 45.009 21.325C44.6257 21.5583 44.1423 21.675 43.559 21.675C43.009 21.675 42.534 21.5583 42.134 21.325C41.7507 21.075 41.4257 20.7583 41.159 20.375C40.909 19.975 40.7257 19.525 40.609 19.025C40.4923 18.525 40.434 18.025 40.434 17.525C40.434 16.9917 40.484 16.4833 40.584 16C40.7007 15.5 40.8757 15.0583 41.109 14.675C41.359 14.2917 41.684 13.9833 42.084 13.75C42.484 13.5167 42.9757 13.4 43.559 13.4C44.1423 13.4 44.6257 13.5167 45.009 13.75C45.3923 13.9833 45.7007 14.2917 45.934 14.675C46.184 15.0417 46.359 15.475 46.459 15.975C46.559 16.4583 46.609 16.9667 46.609 17.5ZM46.659 22.35V24H50.034V6.15H46.484V12.65H46.434C46.034 12.0167 45.484 11.5417 44.784 11.225C44.1007 10.8917 43.3757 10.725 42.609 10.725C41.659 10.725 40.8257 10.9167 40.109 11.3C39.3923 11.6667 38.7923 12.1583 38.309 12.775C37.8423 13.3917 37.484 14.1083 37.234 14.925C37.0007 15.725 36.884 16.5583 36.884 17.425C36.884 18.325 37.0007 19.1917 37.234 20.025C37.484 20.8583 37.8423 21.6 38.309 22.25C38.7923 22.8833 39.4007 23.3917 40.134 23.775C40.8673 24.1417 41.7173 24.325 42.684 24.325C43.534 24.325 44.2923 24.175 44.959 23.875C45.6423 23.5583 46.1923 23.05 46.609 22.35H46.659ZM61.4172 16.15H55.6422C55.6589 15.9 55.7089 15.6167 55.7922 15.3C55.8922 14.9833 56.0505 14.6833 56.2672 14.4C56.5005 14.1167 56.8005 13.8833 57.1672 13.7C57.5505 13.5 58.0255 13.4 58.5922 13.4C59.4589 13.4 60.1005 13.6333 60.5172 14.1C60.9505 14.5667 61.2505 15.25 61.4172 16.15ZM55.6422 18.4H64.9672C65.0339 17.4 64.9505 16.4417 64.7172 15.525C64.4839 14.6083 64.1005 13.7917 63.5672 13.075C63.0505 12.3583 62.3839 11.7917 61.5672 11.375C60.7505 10.9417 59.7922 10.725 58.6922 10.725C57.7089 10.725 56.8089 10.9 55.9922 11.25C55.1922 11.6 54.5005 12.0833 53.9172 12.7C53.3339 13.3 52.8839 14.0167 52.5672 14.85C52.2505 15.6833 52.0922 16.5833 52.0922 17.55C52.0922 18.55 52.2422 19.4667 52.5422 20.3C52.8589 21.1333 53.3005 21.85 53.8672 22.45C54.4339 23.05 55.1255 23.5167 55.9422 23.85C56.7589 24.1667 57.6755 24.325 58.6922 24.325C60.1589 24.325 61.4089 23.9917 62.4422 23.325C63.4755 22.6583 64.2422 21.55 64.7422 20H61.6172C61.5005 20.4 61.1839 20.7833 60.6672 21.15C60.1505 21.5 59.5339 21.675 58.8172 21.675C57.8172 21.675 57.0505 21.4167 56.5172 20.9C55.9839 20.3833 55.6922 19.55 55.6422 18.4ZM75.8727 15.625H79.3477C79.2977 14.7917 79.0977 14.075 78.7477 13.475C78.3977 12.8583 77.9393 12.35 77.3727 11.95C76.8227 11.5333 76.1893 11.225 75.4727 11.025C74.7727 10.825 74.0393 10.725 73.2727 10.725C72.2227 10.725 71.2893 10.9 70.4727 11.25C69.656 11.6 68.9643 12.0917 68.3977 12.725C67.831 13.3417 67.3977 14.0833 67.0977 14.95C66.8143 15.8 66.6727 16.725 66.6727 17.725C66.6727 18.6917 66.831 19.5833 67.1477 20.4C67.4643 21.2 67.906 21.8917 68.4727 22.475C69.0393 23.0583 69.7227 23.5167 70.5227 23.85C71.3393 24.1667 72.231 24.325 73.1977 24.325C74.9143 24.325 76.3227 23.875 77.4227 22.975C78.5227 22.075 79.1893 20.7667 79.4227 19.05H75.9977C75.881 19.85 75.5893 20.4917 75.1227 20.975C74.6727 21.4417 74.0227 21.675 73.1727 21.675C72.6227 21.675 72.156 21.55 71.7727 21.3C71.3893 21.05 71.081 20.7333 70.8477 20.35C70.631 19.95 70.4727 19.5083 70.3727 19.025C70.2727 18.5417 70.2227 18.0667 70.2227 17.6C70.2227 17.1167 70.2727 16.6333 70.3727 16.15C70.4727 15.65 70.6393 15.2 70.8727 14.8C71.1227 14.3833 71.4393 14.05 71.8227 13.8C72.206 13.5333 72.681 13.4 73.2477 13.4C74.7643 13.4 75.6393 14.1417 75.8727 15.625ZM81.7531 6.15V24H85.3031V19.55L86.6781 18.225L90.2281 24H94.5281L89.1031 15.825L93.9781 11.075H89.7781L85.3031 15.725V6.15H81.7531Z",
    transform: "translate(0 -6)",
    fill: "white"
  }));
};

var _excluded = ["children", "onClose", "isOpen", "showAttribution", "className", "style"];
var Modal = /*#__PURE__*/React.forwardRef(function Modal(props, ref) {
  var children = props.children,
      onClose = props.onClose,
      isOpen = props.isOpen,
      showAttribution = props.showAttribution,
      _props$className = props.className,
      className = _props$className === void 0 ? '' : _props$className,
      _props$style = props.style,
      style = _props$style === void 0 ? {} : _props$style,
      other = _objectWithoutPropertiesLoose(props, _excluded);

  var _useState = useState(false),
      mounted = _useState[0],
      setMounted = _useState[1];

  useEffect(function () {
    setMounted(true);
  }, []);
  var modalComponent = React.createElement(Transition, {
    show: isOpen
  }, React.createElement(Transition.Child, {
    enter: "transition ease-out duration-150",
    enterFrom: "opacity-0",
    enterTo: "opacity-100",
    leave: "transition ease-in duration-150",
    leaveFrom: "opacity-100",
    leaveTo: "opacity-0"
  }, React.createElement("div", {
    className: "fixed inset-0 z-40 flex items-end bg-gray-400 bg-opacity-75 dark:bg-gray-600 sm:items-center sm:justify-center",
    "data-testid": "backdrop",
    onClick: onClose
  }, showAttribution ? React.createElement("a", {
    className: "absolute flex text-sm text-center text-gray-100 bottom-5 lg:left-5 xl:left-6 xl:bottom-6",
    href: "https://apideck.com/products/unify",
    target: "_blank",
    rel: "noopener noreferrer"
  }, "Powered by ", React.createElement(Logo, {
    className: "h-5"
  })) : null, React.createElement(Transition.Child, Object.assign({
    enter: "transition ease-out duration-150",
    enterFrom: "opacity-0 transform translate-y-1/4 scale-95",
    enterTo: "opacity-100 scale-100",
    leave: "transition ease-in duration-150",
    leaveFrom: "opacity-100 scale-100",
    leaveTo: "opacity-0  transform translate-y-1/4 scale-95",
    className: "w-full p-5 overflow-hidden bg-white dark:bg-gray-800 dark-text-gray-400 shadow-lg rounded-t-lg sm:p-6 no-scrollbar sm:rounded-lg sm:m-4 sm:max-w-xl relative " + className,
    style: _extends({
      maxHeight: '90%'
    }, style),
    ref: ref,
    role: "dialog",
    id: "modal-component",
    onClick: function onClick(e) {
      return e.stopPropagation();
    }
  }, other), children))));
  return mounted ? createPortal(modalComponent, document.body) : null;
});

function createCommonjsModule(fn, module) {
	return module = { exports: {} }, fn(module, module.exports), module.exports;
}

var runtime_1 = createCommonjsModule(function (module) {
/**
 * Copyright (c) 2014-present, Facebook, Inc.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */

var runtime = (function (exports) {

  var Op = Object.prototype;
  var hasOwn = Op.hasOwnProperty;
  var undefined$1; // More compressible than void 0.
  var $Symbol = typeof Symbol === "function" ? Symbol : {};
  var iteratorSymbol = $Symbol.iterator || "@@iterator";
  var asyncIteratorSymbol = $Symbol.asyncIterator || "@@asyncIterator";
  var toStringTagSymbol = $Symbol.toStringTag || "@@toStringTag";

  function define(obj, key, value) {
    Object.defineProperty(obj, key, {
      value: value,
      enumerable: true,
      configurable: true,
      writable: true
    });
    return obj[key];
  }
  try {
    // IE 8 has a broken Object.defineProperty that only works on DOM objects.
    define({}, "");
  } catch (err) {
    define = function(obj, key, value) {
      return obj[key] = value;
    };
  }

  function wrap(innerFn, outerFn, self, tryLocsList) {
    // If outerFn provided and outerFn.prototype is a Generator, then outerFn.prototype instanceof Generator.
    var protoGenerator = outerFn && outerFn.prototype instanceof Generator ? outerFn : Generator;
    var generator = Object.create(protoGenerator.prototype);
    var context = new Context(tryLocsList || []);

    // The ._invoke method unifies the implementations of the .next,
    // .throw, and .return methods.
    generator._invoke = makeInvokeMethod(innerFn, self, context);

    return generator;
  }
  exports.wrap = wrap;

  // Try/catch helper to minimize deoptimizations. Returns a completion
  // record like context.tryEntries[i].completion. This interface could
  // have been (and was previously) designed to take a closure to be
  // invoked without arguments, but in all the cases we care about we
  // already have an existing method we want to call, so there's no need
  // to create a new function object. We can even get away with assuming
  // the method takes exactly one argument, since that happens to be true
  // in every case, so we don't have to touch the arguments object. The
  // only additional allocation required is the completion record, which
  // has a stable shape and so hopefully should be cheap to allocate.
  function tryCatch(fn, obj, arg) {
    try {
      return { type: "normal", arg: fn.call(obj, arg) };
    } catch (err) {
      return { type: "throw", arg: err };
    }
  }

  var GenStateSuspendedStart = "suspendedStart";
  var GenStateSuspendedYield = "suspendedYield";
  var GenStateExecuting = "executing";
  var GenStateCompleted = "completed";

  // Returning this object from the innerFn has the same effect as
  // breaking out of the dispatch switch statement.
  var ContinueSentinel = {};

  // Dummy constructor functions that we use as the .constructor and
  // .constructor.prototype properties for functions that return Generator
  // objects. For full spec compliance, you may wish to configure your
  // minifier not to mangle the names of these two functions.
  function Generator() {}
  function GeneratorFunction() {}
  function GeneratorFunctionPrototype() {}

  // This is a polyfill for %IteratorPrototype% for environments that
  // don't natively support it.
  var IteratorPrototype = {};
  define(IteratorPrototype, iteratorSymbol, function () {
    return this;
  });

  var getProto = Object.getPrototypeOf;
  var NativeIteratorPrototype = getProto && getProto(getProto(values([])));
  if (NativeIteratorPrototype &&
      NativeIteratorPrototype !== Op &&
      hasOwn.call(NativeIteratorPrototype, iteratorSymbol)) {
    // This environment has a native %IteratorPrototype%; use it instead
    // of the polyfill.
    IteratorPrototype = NativeIteratorPrototype;
  }

  var Gp = GeneratorFunctionPrototype.prototype =
    Generator.prototype = Object.create(IteratorPrototype);
  GeneratorFunction.prototype = GeneratorFunctionPrototype;
  define(Gp, "constructor", GeneratorFunctionPrototype);
  define(GeneratorFunctionPrototype, "constructor", GeneratorFunction);
  GeneratorFunction.displayName = define(
    GeneratorFunctionPrototype,
    toStringTagSymbol,
    "GeneratorFunction"
  );

  // Helper for defining the .next, .throw, and .return methods of the
  // Iterator interface in terms of a single ._invoke method.
  function defineIteratorMethods(prototype) {
    ["next", "throw", "return"].forEach(function(method) {
      define(prototype, method, function(arg) {
        return this._invoke(method, arg);
      });
    });
  }

  exports.isGeneratorFunction = function(genFun) {
    var ctor = typeof genFun === "function" && genFun.constructor;
    return ctor
      ? ctor === GeneratorFunction ||
        // For the native GeneratorFunction constructor, the best we can
        // do is to check its .name property.
        (ctor.displayName || ctor.name) === "GeneratorFunction"
      : false;
  };

  exports.mark = function(genFun) {
    if (Object.setPrototypeOf) {
      Object.setPrototypeOf(genFun, GeneratorFunctionPrototype);
    } else {
      genFun.__proto__ = GeneratorFunctionPrototype;
      define(genFun, toStringTagSymbol, "GeneratorFunction");
    }
    genFun.prototype = Object.create(Gp);
    return genFun;
  };

  // Within the body of any async function, `await x` is transformed to
  // `yield regeneratorRuntime.awrap(x)`, so that the runtime can test
  // `hasOwn.call(value, "__await")` to determine if the yielded value is
  // meant to be awaited.
  exports.awrap = function(arg) {
    return { __await: arg };
  };

  function AsyncIterator(generator, PromiseImpl) {
    function invoke(method, arg, resolve, reject) {
      var record = tryCatch(generator[method], generator, arg);
      if (record.type === "throw") {
        reject(record.arg);
      } else {
        var result = record.arg;
        var value = result.value;
        if (value &&
            typeof value === "object" &&
            hasOwn.call(value, "__await")) {
          return PromiseImpl.resolve(value.__await).then(function(value) {
            invoke("next", value, resolve, reject);
          }, function(err) {
            invoke("throw", err, resolve, reject);
          });
        }

        return PromiseImpl.resolve(value).then(function(unwrapped) {
          // When a yielded Promise is resolved, its final value becomes
          // the .value of the Promise<{value,done}> result for the
          // current iteration.
          result.value = unwrapped;
          resolve(result);
        }, function(error) {
          // If a rejected Promise was yielded, throw the rejection back
          // into the async generator function so it can be handled there.
          return invoke("throw", error, resolve, reject);
        });
      }
    }

    var previousPromise;

    function enqueue(method, arg) {
      function callInvokeWithMethodAndArg() {
        return new PromiseImpl(function(resolve, reject) {
          invoke(method, arg, resolve, reject);
        });
      }

      return previousPromise =
        // If enqueue has been called before, then we want to wait until
        // all previous Promises have been resolved before calling invoke,
        // so that results are always delivered in the correct order. If
        // enqueue has not been called before, then it is important to
        // call invoke immediately, without waiting on a callback to fire,
        // so that the async generator function has the opportunity to do
        // any necessary setup in a predictable way. This predictability
        // is why the Promise constructor synchronously invokes its
        // executor callback, and why async functions synchronously
        // execute code before the first await. Since we implement simple
        // async functions in terms of async generators, it is especially
        // important to get this right, even though it requires care.
        previousPromise ? previousPromise.then(
          callInvokeWithMethodAndArg,
          // Avoid propagating failures to Promises returned by later
          // invocations of the iterator.
          callInvokeWithMethodAndArg
        ) : callInvokeWithMethodAndArg();
    }

    // Define the unified helper method that is used to implement .next,
    // .throw, and .return (see defineIteratorMethods).
    this._invoke = enqueue;
  }

  defineIteratorMethods(AsyncIterator.prototype);
  define(AsyncIterator.prototype, asyncIteratorSymbol, function () {
    return this;
  });
  exports.AsyncIterator = AsyncIterator;

  // Note that simple async functions are implemented on top of
  // AsyncIterator objects; they just return a Promise for the value of
  // the final result produced by the iterator.
  exports.async = function(innerFn, outerFn, self, tryLocsList, PromiseImpl) {
    if (PromiseImpl === void 0) PromiseImpl = Promise;

    var iter = new AsyncIterator(
      wrap(innerFn, outerFn, self, tryLocsList),
      PromiseImpl
    );

    return exports.isGeneratorFunction(outerFn)
      ? iter // If outerFn is a generator, return the full iterator.
      : iter.next().then(function(result) {
          return result.done ? result.value : iter.next();
        });
  };

  function makeInvokeMethod(innerFn, self, context) {
    var state = GenStateSuspendedStart;

    return function invoke(method, arg) {
      if (state === GenStateExecuting) {
        throw new Error("Generator is already running");
      }

      if (state === GenStateCompleted) {
        if (method === "throw") {
          throw arg;
        }

        // Be forgiving, per 25.3.3.3.3 of the spec:
        // https://people.mozilla.org/~jorendorff/es6-draft.html#sec-generatorresume
        return doneResult();
      }

      context.method = method;
      context.arg = arg;

      while (true) {
        var delegate = context.delegate;
        if (delegate) {
          var delegateResult = maybeInvokeDelegate(delegate, context);
          if (delegateResult) {
            if (delegateResult === ContinueSentinel) continue;
            return delegateResult;
          }
        }

        if (context.method === "next") {
          // Setting context._sent for legacy support of Babel's
          // function.sent implementation.
          context.sent = context._sent = context.arg;

        } else if (context.method === "throw") {
          if (state === GenStateSuspendedStart) {
            state = GenStateCompleted;
            throw context.arg;
          }

          context.dispatchException(context.arg);

        } else if (context.method === "return") {
          context.abrupt("return", context.arg);
        }

        state = GenStateExecuting;

        var record = tryCatch(innerFn, self, context);
        if (record.type === "normal") {
          // If an exception is thrown from innerFn, we leave state ===
          // GenStateExecuting and loop back for another invocation.
          state = context.done
            ? GenStateCompleted
            : GenStateSuspendedYield;

          if (record.arg === ContinueSentinel) {
            continue;
          }

          return {
            value: record.arg,
            done: context.done
          };

        } else if (record.type === "throw") {
          state = GenStateCompleted;
          // Dispatch the exception by looping back around to the
          // context.dispatchException(context.arg) call above.
          context.method = "throw";
          context.arg = record.arg;
        }
      }
    };
  }

  // Call delegate.iterator[context.method](context.arg) and handle the
  // result, either by returning a { value, done } result from the
  // delegate iterator, or by modifying context.method and context.arg,
  // setting context.delegate to null, and returning the ContinueSentinel.
  function maybeInvokeDelegate(delegate, context) {
    var method = delegate.iterator[context.method];
    if (method === undefined$1) {
      // A .throw or .return when the delegate iterator has no .throw
      // method always terminates the yield* loop.
      context.delegate = null;

      if (context.method === "throw") {
        // Note: ["return"] must be used for ES3 parsing compatibility.
        if (delegate.iterator["return"]) {
          // If the delegate iterator has a return method, give it a
          // chance to clean up.
          context.method = "return";
          context.arg = undefined$1;
          maybeInvokeDelegate(delegate, context);

          if (context.method === "throw") {
            // If maybeInvokeDelegate(context) changed context.method from
            // "return" to "throw", let that override the TypeError below.
            return ContinueSentinel;
          }
        }

        context.method = "throw";
        context.arg = new TypeError(
          "The iterator does not provide a 'throw' method");
      }

      return ContinueSentinel;
    }

    var record = tryCatch(method, delegate.iterator, context.arg);

    if (record.type === "throw") {
      context.method = "throw";
      context.arg = record.arg;
      context.delegate = null;
      return ContinueSentinel;
    }

    var info = record.arg;

    if (! info) {
      context.method = "throw";
      context.arg = new TypeError("iterator result is not an object");
      context.delegate = null;
      return ContinueSentinel;
    }

    if (info.done) {
      // Assign the result of the finished delegate to the temporary
      // variable specified by delegate.resultName (see delegateYield).
      context[delegate.resultName] = info.value;

      // Resume execution at the desired location (see delegateYield).
      context.next = delegate.nextLoc;

      // If context.method was "throw" but the delegate handled the
      // exception, let the outer generator proceed normally. If
      // context.method was "next", forget context.arg since it has been
      // "consumed" by the delegate iterator. If context.method was
      // "return", allow the original .return call to continue in the
      // outer generator.
      if (context.method !== "return") {
        context.method = "next";
        context.arg = undefined$1;
      }

    } else {
      // Re-yield the result returned by the delegate method.
      return info;
    }

    // The delegate iterator is finished, so forget it and continue with
    // the outer generator.
    context.delegate = null;
    return ContinueSentinel;
  }

  // Define Generator.prototype.{next,throw,return} in terms of the
  // unified ._invoke helper method.
  defineIteratorMethods(Gp);

  define(Gp, toStringTagSymbol, "Generator");

  // A Generator should always return itself as the iterator object when the
  // @@iterator function is called on it. Some browsers' implementations of the
  // iterator prototype chain incorrectly implement this, causing the Generator
  // object to not be returned from this call. This ensures that doesn't happen.
  // See https://github.com/facebook/regenerator/issues/274 for more details.
  define(Gp, iteratorSymbol, function() {
    return this;
  });

  define(Gp, "toString", function() {
    return "[object Generator]";
  });

  function pushTryEntry(locs) {
    var entry = { tryLoc: locs[0] };

    if (1 in locs) {
      entry.catchLoc = locs[1];
    }

    if (2 in locs) {
      entry.finallyLoc = locs[2];
      entry.afterLoc = locs[3];
    }

    this.tryEntries.push(entry);
  }

  function resetTryEntry(entry) {
    var record = entry.completion || {};
    record.type = "normal";
    delete record.arg;
    entry.completion = record;
  }

  function Context(tryLocsList) {
    // The root entry object (effectively a try statement without a catch
    // or a finally block) gives us a place to store values thrown from
    // locations where there is no enclosing try statement.
    this.tryEntries = [{ tryLoc: "root" }];
    tryLocsList.forEach(pushTryEntry, this);
    this.reset(true);
  }

  exports.keys = function(object) {
    var keys = [];
    for (var key in object) {
      keys.push(key);
    }
    keys.reverse();

    // Rather than returning an object with a next method, we keep
    // things simple and return the next function itself.
    return function next() {
      while (keys.length) {
        var key = keys.pop();
        if (key in object) {
          next.value = key;
          next.done = false;
          return next;
        }
      }

      // To avoid creating an additional object, we just hang the .value
      // and .done properties off the next function object itself. This
      // also ensures that the minifier will not anonymize the function.
      next.done = true;
      return next;
    };
  };

  function values(iterable) {
    if (iterable) {
      var iteratorMethod = iterable[iteratorSymbol];
      if (iteratorMethod) {
        return iteratorMethod.call(iterable);
      }

      if (typeof iterable.next === "function") {
        return iterable;
      }

      if (!isNaN(iterable.length)) {
        var i = -1, next = function next() {
          while (++i < iterable.length) {
            if (hasOwn.call(iterable, i)) {
              next.value = iterable[i];
              next.done = false;
              return next;
            }
          }

          next.value = undefined$1;
          next.done = true;

          return next;
        };

        return next.next = next;
      }
    }

    // Return an iterator with no values.
    return { next: doneResult };
  }
  exports.values = values;

  function doneResult() {
    return { value: undefined$1, done: true };
  }

  Context.prototype = {
    constructor: Context,

    reset: function(skipTempReset) {
      this.prev = 0;
      this.next = 0;
      // Resetting context._sent for legacy support of Babel's
      // function.sent implementation.
      this.sent = this._sent = undefined$1;
      this.done = false;
      this.delegate = null;

      this.method = "next";
      this.arg = undefined$1;

      this.tryEntries.forEach(resetTryEntry);

      if (!skipTempReset) {
        for (var name in this) {
          // Not sure about the optimal order of these conditions:
          if (name.charAt(0) === "t" &&
              hasOwn.call(this, name) &&
              !isNaN(+name.slice(1))) {
            this[name] = undefined$1;
          }
        }
      }
    },

    stop: function() {
      this.done = true;

      var rootEntry = this.tryEntries[0];
      var rootRecord = rootEntry.completion;
      if (rootRecord.type === "throw") {
        throw rootRecord.arg;
      }

      return this.rval;
    },

    dispatchException: function(exception) {
      if (this.done) {
        throw exception;
      }

      var context = this;
      function handle(loc, caught) {
        record.type = "throw";
        record.arg = exception;
        context.next = loc;

        if (caught) {
          // If the dispatched exception was caught by a catch block,
          // then let that catch block handle the exception normally.
          context.method = "next";
          context.arg = undefined$1;
        }

        return !! caught;
      }

      for (var i = this.tryEntries.length - 1; i >= 0; --i) {
        var entry = this.tryEntries[i];
        var record = entry.completion;

        if (entry.tryLoc === "root") {
          // Exception thrown outside of any try block that could handle
          // it, so set the completion value of the entire function to
          // throw the exception.
          return handle("end");
        }

        if (entry.tryLoc <= this.prev) {
          var hasCatch = hasOwn.call(entry, "catchLoc");
          var hasFinally = hasOwn.call(entry, "finallyLoc");

          if (hasCatch && hasFinally) {
            if (this.prev < entry.catchLoc) {
              return handle(entry.catchLoc, true);
            } else if (this.prev < entry.finallyLoc) {
              return handle(entry.finallyLoc);
            }

          } else if (hasCatch) {
            if (this.prev < entry.catchLoc) {
              return handle(entry.catchLoc, true);
            }

          } else if (hasFinally) {
            if (this.prev < entry.finallyLoc) {
              return handle(entry.finallyLoc);
            }

          } else {
            throw new Error("try statement without catch or finally");
          }
        }
      }
    },

    abrupt: function(type, arg) {
      for (var i = this.tryEntries.length - 1; i >= 0; --i) {
        var entry = this.tryEntries[i];
        if (entry.tryLoc <= this.prev &&
            hasOwn.call(entry, "finallyLoc") &&
            this.prev < entry.finallyLoc) {
          var finallyEntry = entry;
          break;
        }
      }

      if (finallyEntry &&
          (type === "break" ||
           type === "continue") &&
          finallyEntry.tryLoc <= arg &&
          arg <= finallyEntry.finallyLoc) {
        // Ignore the finally entry if control is not jumping to a
        // location outside the try/catch block.
        finallyEntry = null;
      }

      var record = finallyEntry ? finallyEntry.completion : {};
      record.type = type;
      record.arg = arg;

      if (finallyEntry) {
        this.method = "next";
        this.next = finallyEntry.finallyLoc;
        return ContinueSentinel;
      }

      return this.complete(record);
    },

    complete: function(record, afterLoc) {
      if (record.type === "throw") {
        throw record.arg;
      }

      if (record.type === "break" ||
          record.type === "continue") {
        this.next = record.arg;
      } else if (record.type === "return") {
        this.rval = this.arg = record.arg;
        this.method = "return";
        this.next = "end";
      } else if (record.type === "normal" && afterLoc) {
        this.next = afterLoc;
      }

      return ContinueSentinel;
    },

    finish: function(finallyLoc) {
      for (var i = this.tryEntries.length - 1; i >= 0; --i) {
        var entry = this.tryEntries[i];
        if (entry.finallyLoc === finallyLoc) {
          this.complete(entry.completion, entry.afterLoc);
          resetTryEntry(entry);
          return ContinueSentinel;
        }
      }
    },

    "catch": function(tryLoc) {
      for (var i = this.tryEntries.length - 1; i >= 0; --i) {
        var entry = this.tryEntries[i];
        if (entry.tryLoc === tryLoc) {
          var record = entry.completion;
          if (record.type === "throw") {
            var thrown = record.arg;
            resetTryEntry(entry);
          }
          return thrown;
        }
      }

      // The context.catch method must only be called with a location
      // argument that corresponds to a known catch block.
      throw new Error("illegal catch attempt");
    },

    delegateYield: function(iterable, resultName, nextLoc) {
      this.delegate = {
        iterator: values(iterable),
        resultName: resultName,
        nextLoc: nextLoc
      };

      if (this.method === "next") {
        // Deliberately forget the last sent value so that we don't
        // accidentally pass it on to the delegate.
        this.arg = undefined$1;
      }

      return ContinueSentinel;
    }
  };

  // Regardless of whether this script is executing as a CommonJS module
  // or not, return the runtime object so that we can declare the variable
  // regeneratorRuntime in the outer scope, which allows this module to be
  // injected easily by `bin/regenerator --include-runtime script.js`.
  return exports;

}(
  // If this script is executing as a CommonJS module, use module.exports
  // as the regeneratorRuntime namespace. Otherwise create a new empty
  // object. Either way, the resulting object will be used to initialize
  // the regeneratorRuntime variable at the top of this file.
   module.exports 
));

try {
  regeneratorRuntime = runtime;
} catch (accidentalStrictMode) {
  // This module should not be running in strict mode, so the above
  // assignment should always work unless something is misconfigured. Just
  // in case runtime.js accidentally runs in strict mode, in modern engines
  // we can explicitly access globalThis. In older engines we can escape
  // strict mode using a global Function call. This could conceivably fail
  // if a Content Security Policy forbids using Function, but in that case
  // the proper solution is to fix the accidental strict mode problem. If
  // you've misconfigured your bundler to force strict mode and applied a
  // CSP to forbid Function, and you're not willing to fix either of those
  // problems, please detail your unique predicament in a GitHub issue.
  if (typeof globalThis === "object") {
    globalThis.regeneratorRuntime = runtime;
  } else {
    Function("r", "regeneratorRuntime = r")(runtime);
  }
}
});

var Breadcrumbs = function Breadcrumbs(_ref) {
  var folders = _ref.folders,
      handleClick = _ref.handleClick;
  return React.createElement(Fragment, null, React.createElement("div", {
    className: "flex items-center text-sm"
  }, React.createElement("button", {
    className: "text-gray-600 hover:text-gray-900",
    onClick: function onClick() {
      return handleClick();
    }
  }, React.createElement("svg", {
    xmlns: "http://www.w3.org/2000/svg",
    className: "inline-block w-5 h-5 text-yellow-300",
    fill: "none",
    viewBox: "0 0 24 24",
    stroke: "currentColor"
  }, React.createElement("path", {
    strokeLinecap: "round",
    strokeLinejoin: "round",
    strokeWidth: 2,
    d: "M5 19a2 2 0 01-2-2V7a2 2 0 012-2h4l2 2h4a2 2 0 012 2v1M5 19h14a2 2 0 002-2v-5a2 2 0 00-2-2H9a2 2 0 00-2 2v5a2 2 0 01-2 2z"
  }))), folders.map(function (folder, i) {
    return React.createElement(Transition, {
      show: true,
      enter: "transition ease-out duration-200",
      enterFrom: "transform opacity-60",
      enterTo: "transform opacity-100",
      leave: "transition ease-in duration-200",
      leaveFrom: "transform opacity-100",
      leaveTo: "transform opacity-0",
      key: i
    }, React.createElement(Transition.Child, {
      enter: "transition ease-out duration-200",
      enterFrom: "transform opacity-60 translate-x-1",
      enterTo: "transform opacity-100",
      leave: "transition ease-in duration-100",
      leaveFrom: "transform opacity-100",
      leaveTo: "transform opacity-0",
      className: "flex items-center",
      as: "div",
      key: folder.id
    }, React.createElement("svg", {
      className: "flex-shrink-0 w-5 h-5 text-gray-300",
      fill: "currentColor",
      viewBox: "0 0 20 20"
    }, React.createElement("path", {
      fillRule: "evenodd",
      d: "M7.293 14.707a1 1 0 010-1.414L10.586 10 7.293 6.707a1 1 0 011.414-1.414l4 4a1 1 0 010 1.414l-4 4a1 1 0 01-1.414 0z",
      clipRule: "evenodd"
    })), i === folders.length - 1 ? React.createElement("span", {
      className: "font-medium text-gray-900"
    }, folder.name) : React.createElement("button", {
      className: "text-gray-600 hover:text-gray-900",
      onClick: function onClick() {
        return handleClick(folder);
      }
    }, folder.name)));
  })));
};

var formatBytes = function formatBytes(bytes, decimals) {
  if (decimals === void 0) {
    decimals = 2;
  }

  if (bytes === 0) return '0 Bytes';
  var k = 1024;
  var dm = decimals < 0 ? 0 : decimals;
  var sizes = ['Bytes', 'KB', 'MB', 'GB', 'TB', 'PB', 'EB', 'ZB', 'YB'];
  var i = Math.floor(Math.log(bytes) / Math.log(k));
  return parseFloat((bytes / Math.pow(k, i)).toFixed(dm)) + ' ' + sizes[i];
};

var FileDetails = function FileDetails(_ref) {
  var _file$owner, _file$owner2;

  var file = _ref.file,
      setFile = _ref.setFile,
      onSelect = _ref.onSelect;
  if (!file) return React.createElement("p", {
    className: "p-4 text-center"
  }, "No file found");
  return React.createElement("div", {
    className: "flex flex-col justify-between h-full"
  }, React.createElement("dl", {
    className: "grid grid-cols-2 gap-x-4 gap-y-6"
  }, React.createElement("div", {
    className: "col-span-1"
  }, React.createElement("dt", {
    className: "text-sm font-medium text-gray-500"
  }, "Name"), React.createElement("dd", {
    className: "mt-1 text-sm text-gray-900"
  }, file == null ? void 0 : file.name)), React.createElement("div", {
    className: "col-span-1"
  }, React.createElement("dt", {
    className: "text-sm font-medium text-gray-500"
  }, "Size"), React.createElement("dd", {
    className: "mt-1 text-sm text-gray-900"
  }, file != null && file.size ? formatBytes(file == null ? void 0 : file.size) : '-')), React.createElement("div", {
    className: "col-span-1"
  }, React.createElement("dt", {
    className: "text-sm font-medium text-gray-500"
  }, "Mime type"), React.createElement("dd", {
    className: "mt-1 text-sm text-gray-900"
  }, file == null ? void 0 : file.mime_type)), React.createElement("div", {
    className: "col-span-1"
  }, React.createElement("dt", {
    className: "text-sm font-medium text-gray-500"
  }, "Downloadable"), React.createElement("dd", {
    className: "mt-1 text-sm text-gray-900"
  }, file != null && file.downloadable ? 'Yes' : 'No')), React.createElement("div", {
    className: "col-span-1"
  }, React.createElement("dt", {
    className: "text-sm font-medium text-gray-500"
  }, "Created"), React.createElement("dd", {
    className: "mt-1 text-sm text-gray-900"
  }, React.createElement("div", null, React.createElement("span", {
    className: "mr-2 text-gray-900"
  }, (file == null ? void 0 : file.created_at) && new Date(file.created_at).toLocaleDateString()), React.createElement("span", {
    className: "text-gray-500 "
  }, (file == null ? void 0 : file.created_at) && new Date(file.created_at).toLocaleTimeString())))), React.createElement("div", {
    className: "col-span-1"
  }, React.createElement("dt", {
    className: "text-sm font-medium text-gray-500"
  }, "Updated"), React.createElement("dd", {
    className: "mt-1 text-sm text-gray-900"
  }, React.createElement("div", null, React.createElement("span", {
    className: "mr-2 text-gray-900"
  }, (file == null ? void 0 : file.updated_at) && new Date(file.updated_at).toLocaleDateString()), React.createElement("span", {
    className: "text-gray-500 "
  }, (file == null ? void 0 : file.updated_at) && new Date(file.updated_at).toLocaleTimeString())))), React.createElement("div", {
    className: "col-span-1"
  }, React.createElement("dt", {
    className: "text-sm font-medium text-gray-500"
  }, "Owner"), React.createElement("dd", {
    className: "mt-1 text-sm text-gray-900"
  }, file == null ? void 0 : (_file$owner = file.owner) == null ? void 0 : _file$owner.name)), React.createElement("div", {
    className: "col-span-1"
  }, React.createElement("dt", {
    className: "text-sm font-medium text-gray-500"
  }, "Email"), React.createElement("dd", {
    className: "mt-1 text-sm text-gray-900"
  }, file == null ? void 0 : (_file$owner2 = file.owner) == null ? void 0 : _file$owner2.email))), React.createElement("div", {
    className: "flex justify-between pt-6"
  }, React.createElement("button", {
    type: "button",
    className: "items-center w-full px-3 py-2 mr-2 text-sm font-medium leading-4 text-gray-700 bg-white border border-gray-300 rounded-md shadow-sm sm:mr-3 hover:bg-gray-50 focus:outline-none focus:ring-2 focus:ring-offset-2 focus:ring-indigo-500",
    onClick: function onClick() {
      return setFile(null);
    }
  }, "Close"), React.createElement("button", {
    type: "button",
    className: "items-center w-full px-3 py-2 ml-2 text-sm font-medium leading-4 text-white bg-blue-600 border border-transparent rounded-md shadow-sm sm:ml-3 hover:bg-blue-700 focus:outline-none focus:ring-2 focus:ring-offset-2 focus:ring-indigo-500",
    onClick: function onClick() {
      return onSelect(file);
    }
  }, "Select file")));
};

var LoadingTable = function LoadingTable(_ref) {
  var isSearching = _ref.isSearching;
  var columns = [{
    Header: 'Name'
  }, {
    Header: 'Size'
  }, {
    Header: 'Updated'
  }];
  var array = Array.from(Array(6).keys());

  if (isSearching) {
    columns = [].concat(columns, [{
      Header: 'Service'
    }]);
  }

  return React.createElement("div", {
    className: "overflow-auto"
  }, React.createElement("table", {
    className: "min-w-full divide-y divide-gray-200"
  }, React.createElement("thead", null, React.createElement("tr", null, columns.map(function (column, i) {
    return React.createElement("th", {
      key: "column-" + i,
      className: "py-3 space-x-6 text-xs font-medium tracking-wide text-left text-gray-500 uppercase " + (i === columns.length - 1 ? 'text-right' : '')
    }, column.Header);
  }))), React.createElement("tbody", {
    className: "bg-white divide-y divide-gray-200"
  }, array.map(function (_, i) {
    return React.createElement(LoadingRow, {
      key: i,
      columns: columns
    });
  }))));
};
var LoadingRow = function LoadingRow(_ref2) {
  var columns = _ref2.columns;
  return React.createElement("tr", null, columns.map(function (_, i) {
    return React.createElement("td", {
      className: "py-4 space-x-6 text-sm font-medium truncate whitespace-nowrap " + (i === columns.length - 1 ? 'text-right' : ''),
      key: "cell-" + i,
      style: {
        maxWidth: 80
      }
    }, React.createElement("span", {
      className: "px-16 py-0 bg-gray-200 rounded-sm animate-pulse"
    }));
  }));
};

var months = ['Jan', 'Feb', 'Mar', 'Apr', 'May', 'Jun', 'Jul', 'Aug', 'Sep', 'Oct', 'Nov', 'Dec'];

var FilesTable = function FilesTable(_ref) {
  var _ref$data = _ref.data,
      data = _ref$data === void 0 ? [] : _ref$data,
      isLoadingMore = _ref.isLoadingMore,
      handleSelect = _ref.handleSelect,
      searchMode = _ref.searchMode,
      uploadingMode = _ref.uploadingMode;
  var columns = React.useMemo(function () {
    return [{
      Header: 'Name',
      accessor: 'name',
      Cell: function Cell(props) {
        var type = props.row.original.type;

        if (type === 'folder') {
          return React.createElement("div", {
            className: "flex items-center"
          }, React.createElement("svg", {
            xmlns: "http://www.w3.org/2000/svg",
            className: "inline-block w-5 h-5 mr-1 text-yellow-300",
            fill: "none",
            viewBox: "0 0 24 24",
            stroke: "currentColor"
          }, React.createElement("path", {
            strokeLinecap: "round",
            strokeLinejoin: "round",
            strokeWidth: 2,
            d: "M3 7v10a2 2 0 002 2h14a2 2 0 002-2V9a2 2 0 00-2-2h-6l-2-2H5a2 2 0 00-2 2z"
          })), React.createElement("span", {
            className: "truncate"
          }, props.value));
        }

        return React.createElement("div", {
          className: "flex items-center truncate"
        }, React.createElement("svg", {
          viewBox: "0 0 24 24",
          className: "inline-block w-5 h-5 mr-1 text-gray-500"
        }, React.createElement("path", {
          d: "M17 6v12c0 .52-.2 1-1 1H4c-.7 0-1-.33-1-1V2c0-.55.42-1 1-1h8l5 5zM14 8h-3.13c-.51 0-.87-.34-.87-.87V4",
          stroke: "currentColor",
          fill: "none",
          fillRule: "evenodd",
          strokeLinejoin: "round"
        })), React.createElement("span", {
          className: "truncate"
        }, props.value));
      }
    }, {
      Header: 'Size',
      accessor: 'size',
      Cell: function Cell(_ref2) {
        var value = _ref2.value;
        return React.createElement("div", null, value ? formatBytes(value) : null);
      }
    }, {
      Header: 'Updated',
      accessor: 'updated_at',
      Cell: function Cell(_ref3) {
        var value = _ref3.value;
        if (!value) return React.createElement("span", null);
        var date = new Date(value);
        return React.createElement("span", null, months[date.getMonth()] + " " + date.getDay() + ", " + new Date(value).toLocaleTimeString([], {
          hour: '2-digit',
          minute: '2-digit'
        }));
      }
    }, {
      Header: 'Service',
      accessor: 'connection',
      Cell: function Cell(_ref4) {
        var value = _ref4.value;
        if (!value) return React.createElement("span", {
          className: "hidden text-gray-900"
        }, "-");
        return React.createElement("div", {
          className: "flex justify-end pr-2"
        }, React.createElement("img", {
          className: "inline-block w-5 h-5 text-right rounded-full",
          src: value != null && value.icon ? value.icon : '/img/logo.png',
          alt: value.service_id
        }));
      }
    }];
  }, []);

  var _useTable = useTable({
    columns: columns,
    data: data
  }, useSortBy),
      getTableBodyProps = _useTable.getTableBodyProps,
      headerGroups = _useTable.headerGroups,
      rows = _useTable.rows,
      prepareRow = _useTable.prepareRow;

  return React.createElement("div", {
    className: "overflow-auto"
  }, React.createElement("table", {
    className: "min-w-full divide-y divide-gray-200"
  }, React.createElement("thead", {
    className: ""
  }, headerGroups.map(function (headerGroup, i) {
    return React.createElement("tr", Object.assign({}, headerGroup.getHeaderGroupProps(), {
      key: "headerGroup" + i
    }), headerGroup.headers.map(function (column, i) {
      if (i === headerGroup.headers.length - 1 && !searchMode) {
        return null;
      }

      return React.createElement("th", Object.assign({
        key: "column-" + i,
        className: "py-3 pr-2 space-x-6 text-xs font-medium tracking-wide text-left text-gray-500 uppercase " + (i === headerGroup.headers.length - 2 && !searchMode || i === headerGroup.headers.length - 1 && searchMode ? 'text-right' : '')
      }, column.getHeaderProps(column.getSortByToggleProps())), column.render('Header'), React.createElement("span", null, column.isSorted ? column.isSortedDesc ? React.createElement("svg", {
        xmlns: "http://www.w3.org/2000/svg",
        fill: "none",
        viewBox: "0 0 24 24",
        stroke: "currentColor",
        className: "inline-block w-4 h-4 ml-2 text-gray-500"
      }, React.createElement("path", {
        strokeLinecap: "round",
        strokeLinejoin: "round",
        strokeWidth: 2,
        d: "M19 9l-7 7-7-7"
      })) : React.createElement("svg", {
        xmlns: "http://www.w3.org/2000/svg",
        fill: "none",
        viewBox: "0 0 24 24",
        stroke: "currentColor",
        className: "inline-block w-4 h-4 ml-2 text-gray-500"
      }, React.createElement("path", {
        strokeLinecap: "round",
        strokeLinejoin: "round",
        strokeWidth: 2,
        d: "M5 15l7-7 7 7"
      })) : ''));
    }));
  })), React.createElement("tbody", Object.assign({}, getTableBodyProps(), {
    className: "bg-white divide-y divide-gray-200"
  }), rows.map(function (row, i) {
    prepareRow(row);
    return React.createElement(Transition, Object.assign({
      show: true,
      enter: "transition ease-out duration-300",
      enterFrom: "transform opacity-40",
      enterTo: "transform opacity-100",
      leave: "transition ease-in duration-300",
      leaveFrom: "transform opacity-100",
      leaveTo: "transform opacity-0",
      as: "tr"
    }, row.getRowProps(), {
      key: "row-" + i,
      className: "" + (uploadingMode && row.original.type === 'file' ? 'text-gray-300' : 'cursor-pointer hover:bg-gray-50 text-gray-900'),
      onClick: function onClick() {
        if (!uploadingMode || row.original.type === 'folder') {
          handleSelect(row.original);
        }
      }
    }), row.cells.map(function (cell, i) {
      return React.createElement("td", Object.assign({
        className: "py-3 space-x-6 text-xs truncate max-w-2xs whitespace-nowrap " + (i === row.cells.length - 2 && !searchMode || i === row.cells.length - 1 && searchMode ? 'text-right' : ''),
        style: {
          maxWidth: '16rem'
        }
      }, cell.getCellProps(), {
        key: "cell-" + i
      }), cell.render('Cell'));
    }));
  }))), isLoadingMore ? React.createElement("table", {
    className: "min-w-full divide-y divide-gray-200"
  }, React.createElement("tbody", {
    className: "bg-white divide-y divide-gray-200"
  }, Array.from(Array(12).keys()).map(function (key) {
    return React.createElement(LoadingRow, {
      key: key,
      columns: columns
    });
  }))) : '');
};

var Toast = function Toast(_ref) {
  var title = _ref.title,
      description = _ref.description,
      _ref$type = _ref.type,
      type = _ref$type === void 0 ? 'info' : _ref$type,
      id = _ref.id,
      autoClose = _ref.autoClose,
      closeAfter = _ref.closeAfter,
      closeText = _ref.closeText,
      image = _ref.image;

  var _useState = useState(false),
      shouldShow = _useState[0],
      setShouldShow = _useState[1];

  var _useToast = useToast(),
      removeToast = _useToast.removeToast;

  useEffect(function () {
    if (!shouldShow) setShouldShow(true); // eslint-disable-next-line react-hooks/exhaustive-deps
  }, []);
  useEffect(function () {
    if (autoClose || closeAfter) {
      var timer = setTimeout(function () {
        setShouldShow(false);
        setTimeout(function () {
          removeToast(id);
        }, 300);
      }, closeAfter || 3500);
      return function () {
        clearTimeout(timer);
      };
    }

    return;
  }, [id, removeToast, autoClose]);
  var icon = {
    info: React.createElement("svg", {
      className: "w-6 h-6 text-indigo-400",
      fill: "none",
      viewBox: "0 0 24 24",
      stroke: "currentColor",
      "data-testid": "icon"
    }, React.createElement("path", {
      d: "M13 16H12V12H11M12 8H12.01M21 12C21 16.9706 16.9706 21 12 21C7.02944 21 3 16.9706 3 12C3 7.02944 7.02944 3 12 3C16.9706 3 21 7.02944 21 12Z",
      strokeWidth: "2",
      strokeLinecap: "round",
      strokeLinejoin: "round"
    })),
    success: React.createElement("svg", {
      className: "w-6 h-6 text-green-400",
      fill: "none",
      viewBox: "0 0 24 24",
      stroke: "currentColor",
      "data-testid": "icon"
    }, React.createElement("path", {
      strokeLinecap: "round",
      strokeLinejoin: "round",
      strokeWidth: "2",
      d: "M9 12l2 2 4-4m6 2a9 9 0 11-18 0 9 9 0 0118 0z"
    })),
    warning: React.createElement("svg", {
      className: "w-6 h-6 text-yellow-500",
      fill: "none",
      viewBox: "0 0 24 24",
      stroke: "currentColor",
      "data-testid": "icon"
    }, React.createElement("path", {
      d: "M12 8V12M12 16H12.01M21 12C21 16.9706 16.9706 21 12 21C7.02944 21 3 16.9706 3 12C3 7.02944 7.02944 3 12 3C16.9706 3 21 7.02944 21 12Z",
      strokeWidth: "2",
      strokeLinecap: "round",
      strokeLinejoin: "round"
    })),
    error: React.createElement("svg", {
      className: "w-6 h-6 text-red-400",
      fill: "none",
      viewBox: "0 0 24 24",
      stroke: "currentColor",
      "data-testid": "icon"
    }, React.createElement("path", {
      d: "M12 8V12M12 16H12.01M21 12C21 16.9706 16.9706 21 12 21C7.02944 21 3 16.9706 3 12C3 7.02944 7.02944 3 12 3C16.9706 3 21 7.02944 21 12Z",
      strokeWidth: "2",
      strokeLinecap: "round",
      strokeLinejoin: "round"
    }))
  };
  var borderStyle = {
    success: 'border-green-400',
    warning: 'border-yellow-500',
    error: 'border-red-400',
    info: 'border-indigo-400'
  };
  return React.createElement("div", {
    className: "flex items-end justify-end w-full px-4 pointer-events-none top-3 right-3",
    key: id
  }, React.createElement(Transition, {
    show: shouldShow,
    enter: "transform ease-out duration-300 transition",
    enterFrom: "translate-y-2 opacity-0 sm:translate-y-0 sm:translate-x-2",
    enterTo: "translate-y-0 opacity-100 sm:translate-x-0",
    leave: "transition ease-in duration-200",
    leaveFrom: "opacity-100",
    leaveTo: "opacity-0",
    className: "w-full bg-white dark:bg-gray-800 border-l-2 rounded-md shadow-lg pointer-events-auto " + borderStyle[type],
    style: {
      width: 'calc(100% - 120px)',
      left: 60
    },
    "data-testid": "toast"
  }, React.createElement("div", {
    className: "flex overflow-hidden rounded-lg shadow-xs"
  }, React.createElement("div", {
    className: "flex flex-col justify-center w-full p-3"
  }, React.createElement("div", {
    className: "flex items-start"
  }, React.createElement("div", {
    className: "flex-shrink-0"
  }, icon[type]), React.createElement("div", {
    className: "ml-2 w-0 flex-1 pt-0.5 overflow-hidden"
  }, React.createElement("p", {
    className: "text-sm font-medium leading-5 text-gray-900 dark:text-white"
  }, title), description ? React.createElement("p", {
    className: "mt-1 text-sm leading-5 text-gray-500 dark:text-gray-300"
  }, description) : null, image ? React.createElement("img", {
    src: image,
    className: "mt-2 rounded-md"
  }) : null))), React.createElement("div", {
    className: "flex border-l border-gray-200 group"
  }, React.createElement("button", {
    className: "flex items-center justify-center w-full p-3 text-sm font-medium text-gray-600 border border-transparent rounded-none rounded-r-lg hover:text-gray-500 focus:outline-none focus:ring-1 focus:ring-gray-300",
    onClick: function onClick() {
      return removeToast(id);
    },
    "data-testid": "remove"
  }, closeText ? React.createElement("span", null, closeText) : React.createElement("svg", {
    className: "text-gray-400 fill-current w-7 h-7 group-hover:text-gray-500 dark:text-gray-300 dark:group-hover:text-gray-100",
    xmlns: "http://www.w3.org/2000/svg",
    viewBox: "0 0 24 24",
    width: "22",
    height: "22"
  }, React.createElement("path", {
    d: "M16.24 14.83a1 1 0 0 1-1.41 1.41L12 13.41l-2.83 2.83a1 1 0 0 1-1.41-1.41L10.59 12 7.76 9.17a1 1 0 0 1 1.41-1.41L12 10.59l2.83-2.83a1 1 0 0 1 1.41 1.41L13.41 12l2.83 2.83z"
  })))))));
};

var ToastContext = /*#__PURE__*/createContext({});
var id = 1;
var ToastProvider = function ToastProvider(_ref) {
  var children = _ref.children;

  var _useState = useState([]),
      toasts = _useState[0],
      setToasts = _useState[1];

  var addToast = useCallback(function (toast) {
    setToasts(function (toasts) {
      return [].concat(toasts, [_extends({}, toast, {
        id: id++
      })]);
    });
  }, [setToasts]);
  var removeToast = useCallback(function (id) {
    setToasts(function (toasts) {
      return toasts.filter(function (toast) {
        return toast.id !== id;
      });
    });
  }, [setToasts]);
  return React.createElement(ToastContext.Provider, {
    value: {
      addToast: addToast,
      removeToast: removeToast
    }
  }, React.createElement("div", {
    className: "absolute left-0 right-0 w-full",
    style: {
      zIndex: 100,
      top: 10
    }
  }, toasts == null ? void 0 : toasts.map(function (toast, i) {
    return React.createElement(Toast, Object.assign({
      key: i
    }, toast));
  })), children);
};
var useToast = function useToast() {
  return useContext(ToastContext);
};

var useUpload = function useUpload(_ref) {
  var onSuccess = _ref.onSuccess;

  var _useState = useState(false),
      isLoading = _useState[0],
      setIsLoading = _useState[1];

  var _useState2 = useState(null),
      progress = _useState2[0],
      setProgress = _useState2[1];

  var _useToast = useToast(),
      addToast = _useToast.addToast;

  var uploadFile = /*#__PURE__*/function () {
    var _ref3 = _asyncToGenerator( /*#__PURE__*/runtime_1.mark(function _callee(_ref2) {
      var file, appId, consumerId, serviceId, folderId, jwt, headers;
      return runtime_1.wrap(function _callee$(_context) {
        while (1) {
          switch (_context.prev = _context.next) {
            case 0:
              file = _ref2.file, appId = _ref2.appId, consumerId = _ref2.consumerId, serviceId = _ref2.serviceId, folderId = _ref2.folderId, jwt = _ref2.jwt;
              headers = {
                'x-apideck-auth-type': 'JWT',
                'x-apideck-app-id': appId,
                'x-apideck-consumer-id': consumerId,
                'x-apideck-service-id': serviceId,
                'x-apideck-metadata': JSON.stringify({
                  name: file.name,
                  parent_folder_id: folderId
                }),
                Authorization: "Bearer " + jwt
              };
              setIsLoading(true);
              setProgress(null); // if file bigger then 4mb then use multi part upload

              if (!(file.size > 4000000)) {
                _context.next = 10;
                break;
              }

              setProgress(5);
              _context.next = 8;
              return uploadLargeFile(file, headers, folderId);

            case 8:
              _context.next = 12;
              break;

            case 10:
              _context.next = 12;
              return uploadRegularFile(file, headers);

            case 12:
            case "end":
              return _context.stop();
          }
        }
      }, _callee);
    }));

    return function uploadFile(_x) {
      return _ref3.apply(this, arguments);
    };
  }();

  var uploadRegularFile = /*#__PURE__*/function () {
    var _ref4 = _asyncToGenerator( /*#__PURE__*/runtime_1.mark(function _callee2(file, headers) {
      var raw, response, fileDetails;
      return runtime_1.wrap(function _callee2$(_context2) {
        while (1) {
          switch (_context2.prev = _context2.next) {
            case 0:
              _context2.prev = 0;
              _context2.next = 3;
              return fetch('https://unify.apideck.com/file-storage/files', {
                headers: headers,
                method: 'POST',
                body: file
              });

            case 3:
              raw = _context2.sent;
              _context2.next = 6;
              return raw.json();

            case 6:
              response = _context2.sent;

              if (!response.error) {
                _context2.next = 11;
                break;
              }

              addToast({
                title: 'Something went wrong',
                description: response.message,
                type: 'error',
                autoClose: true
              });
              _context2.next = 16;
              break;

            case 11:
              addToast({
                title: 'File uploaded',
                description: 'File successfully uploaded',
                type: 'success',
                autoClose: true
              });
              _context2.next = 14;
              return getFile(response.data.id, headers);

            case 14:
              fileDetails = _context2.sent;
              onSuccess((fileDetails == null ? void 0 : fileDetails.data) || file);

            case 16:
              _context2.next = 21;
              break;

            case 18:
              _context2.prev = 18;
              _context2.t0 = _context2["catch"](0);
              addToast({
                title: 'Something went wrong',
                description: _context2.t0,
                type: 'error'
              });

            case 21:
              _context2.prev = 21;
              setIsLoading(false);
              return _context2.finish(21);

            case 24:
            case "end":
              return _context2.stop();
          }
        }
      }, _callee2, null, [[0, 18, 21, 24]]);
    }));

    return function uploadRegularFile(_x2, _x3) {
      return _ref4.apply(this, arguments);
    };
  }();

  var getFile = /*#__PURE__*/function () {
    var _ref5 = _asyncToGenerator( /*#__PURE__*/runtime_1.mark(function _callee3(fileId, headers) {
      var raw, response;
      return runtime_1.wrap(function _callee3$(_context3) {
        while (1) {
          switch (_context3.prev = _context3.next) {
            case 0:
              _context3.next = 2;
              return fetch("https://unify.apideck.com/file-storage/files/" + fileId, {
                headers: headers
              });

            case 2:
              raw = _context3.sent;
              _context3.next = 5;
              return raw.json();

            case 5:
              response = _context3.sent;
              return _context3.abrupt("return", response);

            case 7:
            case "end":
              return _context3.stop();
          }
        }
      }, _callee3);
    }));

    return function getFile(_x4, _x5) {
      return _ref5.apply(this, arguments);
    };
  }();

  var getPartSize = /*#__PURE__*/function () {
    var _ref6 = _asyncToGenerator( /*#__PURE__*/runtime_1.mark(function _callee4(file, headers, folderId) {
      var _sessionResponse$data, _session$data, raw, sessionResponse, sessionId, rawSession, session;

      return runtime_1.wrap(function _callee4$(_context4) {
        while (1) {
          switch (_context4.prev = _context4.next) {
            case 0:
              _context4.prev = 0;
              _context4.next = 3;
              return fetch('https://unify.apideck.com/file-storage/upload-sessions', {
                headers: _extends({}, headers, {
                  'Content-Type': 'application/json'
                }),
                method: 'POST',
                body: JSON.stringify({
                  size: file.size,
                  name: file.name,
                  parent_folder_id: folderId
                })
              });

            case 3:
              raw = _context4.sent;
              _context4.next = 6;
              return raw.json();

            case 6:
              sessionResponse = _context4.sent;
              sessionId = (_sessionResponse$data = sessionResponse.data) == null ? void 0 : _sessionResponse$data.id;

              if (sessionId) {
                _context4.next = 10;
                break;
              }

              return _context4.abrupt("return", {
                partSize: null,
                sessionId: null
              });

            case 10:
              _context4.next = 12;
              return fetch("https://unify.apideck.com/file-storage/upload-sessions/" + sessionId, {
                headers: _extends({}, headers, {
                  'Content-Type': 'application/json'
                })
              });

            case 12:
              rawSession = _context4.sent;
              _context4.next = 15;
              return rawSession.json();

            case 15:
              session = _context4.sent;
              return _context4.abrupt("return", {
                partSize: (_session$data = session.data) == null ? void 0 : _session$data.part_size,
                sessionId: sessionId
              });

            case 19:
              _context4.prev = 19;
              _context4.t0 = _context4["catch"](0);
              console.log('Session error', _context4.t0);
              return _context4.abrupt("return", {
                partSize: null,
                sessionId: null
              });

            case 23:
            case "end":
              return _context4.stop();
          }
        }
      }, _callee4, null, [[0, 19]]);
    }));

    return function getPartSize(_x6, _x7, _x8) {
      return _ref6.apply(this, arguments);
    };
  }();

  var uploadLargeFile = /*#__PURE__*/function () {
    var _ref7 = _asyncToGenerator( /*#__PURE__*/runtime_1.mark(function _callee6(file, headers, folderId) {
      var _yield$getPartSize, partSize, sessionId, currentChunkStartByte, currentChunkFinalByte, index, uploadChunk;

      return runtime_1.wrap(function _callee6$(_context6) {
        while (1) {
          switch (_context6.prev = _context6.next) {
            case 0:
              _context6.prev = 0;
              _context6.next = 3;
              return getPartSize(file, headers, folderId);

            case 3:
              _yield$getPartSize = _context6.sent;
              partSize = _yield$getPartSize.partSize;
              sessionId = _yield$getPartSize.sessionId;
              currentChunkStartByte = 0;
              currentChunkFinalByte = partSize > file.size ? file.size : partSize;
              index = 0;

              uploadChunk = /*#__PURE__*/function () {
                var _ref8 = _asyncToGenerator( /*#__PURE__*/runtime_1.mark(function _callee5() {
                  var chunk, raw, response, remainingBytes, result, finalResponse;
                  return runtime_1.wrap(function _callee5$(_context5) {
                    while (1) {
                      switch (_context5.prev = _context5.next) {
                        case 0:
                          chunk = file.slice(currentChunkStartByte, currentChunkFinalByte);
                          _context5.prev = 1;
                          _context5.next = 4;
                          return fetch("https://unify.apideck.com/file-storage/upload-sessions/" + sessionId + "?part_number=" + index, {
                            headers: _extends({}, headers, {
                              'Content-Type': file.type
                            }),
                            method: 'PUT',
                            body: chunk
                          });

                        case 4:
                          raw = _context5.sent;
                          _context5.next = 7;
                          return raw.json();

                        case 7:
                          response = _context5.sent;

                          if (!response.error) {
                            _context5.next = 11;
                            break;
                          }

                          addToast({
                            title: 'Something went wrong',
                            description: response.message,
                            type: 'error'
                          });
                          return _context5.abrupt("return");

                        case 11:
                          remainingBytes = file.size - currentChunkFinalByte;

                          if (!(currentChunkFinalByte === file.size)) {
                            _context5.next = 26;
                            break;
                          }

                          setProgress(100); // All chunks are uploaded. Finish the upload session.

                          _context5.next = 16;
                          return fetch("https://unify.apideck.com/file-storage/upload-sessions/" + sessionId + "/finish", {
                            headers: headers,
                            method: 'POST'
                          });

                        case 16:
                          result = _context5.sent;
                          _context5.next = 19;
                          return result.json();

                        case 19:
                          finalResponse = _context5.sent;
                          setIsLoading(false);
                          addToast({
                            title: 'File uploaded',
                            description: 'File successfully uploaded',
                            type: 'success',
                            autoClose: true
                          });
                          onSuccess(finalResponse.data || file);
                          return _context5.abrupt("return");

                        case 26:
                          if (remainingBytes < partSize) {
                            // Last chunk to upload
                            currentChunkStartByte = currentChunkFinalByte;
                            currentChunkFinalByte = currentChunkStartByte + remainingBytes;
                          } else {
                            // Next chunk to upload
                            currentChunkStartByte = currentChunkFinalByte;
                            currentChunkFinalByte = currentChunkStartByte + partSize;
                          }

                        case 27:
                          index++;
                          setProgress(index / Math.ceil(file.size / partSize) * 100);
                          uploadChunk();
                          _context5.next = 36;
                          break;

                        case 32:
                          _context5.prev = 32;
                          _context5.t0 = _context5["catch"](1);
                          setIsLoading(false);
                          addToast({
                            title: 'Something went wrong',
                            description: _context5.t0,
                            type: 'error'
                          });

                        case 36:
                        case "end":
                          return _context5.stop();
                      }
                    }
                  }, _callee5, null, [[1, 32]]);
                }));

                return function uploadChunk() {
                  return _ref8.apply(this, arguments);
                };
              }();

              console.log("uploading " + Math.ceil(file.size / partSize) + " chunks of " + partSize + " bytes");

              if (sessionId && partSize) {
                uploadChunk();
              }

              _context6.next = 17;
              break;

            case 14:
              _context6.prev = 14;
              _context6.t0 = _context6["catch"](0);
              console.log('Error', _context6.t0);

            case 17:
            case "end":
              return _context6.stop();
          }
        }
      }, _callee6, null, [[0, 14]]);
    }));

    return function uploadLargeFile(_x9, _x10, _x11) {
      return _ref7.apply(this, arguments);
    };
  }();

  return {
    uploadFile: uploadFile,
    isLoading: isLoading,
    progress: progress
  };
};

var SaveFileForm = function SaveFileForm(_ref) {
  var file = _ref.file,
      folderId = _ref.folderId,
      appId = _ref.appId,
      consumerId = _ref.consumerId,
      serviceId = _ref.serviceId,
      jwt = _ref.jwt,
      onSuccess = _ref.onSuccess;

  var _useState = useState(file.name),
      fileName = _useState[0],
      setFileName = _useState[1];

  var _useUpload = useUpload({
    onSuccess: onSuccess
  }),
      uploadFile = _useUpload.uploadFile,
      isLoading = _useUpload.isLoading;

  return React.createElement("div", {
    className: "absolute bottom-0 left-0 right-0 bg-white border-t"
  }, React.createElement("div", {
    className: "flex items-center justify-center p-4"
  }, React.createElement("input", {
    className: "w-full px-3 py-2 text-sm leading-tight text-gray-700 border border-gray-200 rounded shadow-sm appearance-none focus:outline-none focus:shadow-outline",
    name: "fileName",
    placeholder: "File name",
    autoComplete: "off",
    onChange: function onChange(event) {
      return setFileName(event.target.value);
    },
    value: fileName
  }), React.createElement("button", {
    type: "button",
    className: "items-center px-3 py-2 ml-2 text-sm font-medium leading-4 text-white bg-blue-600 border border-transparent rounded-md shadow-sm whitespace-nowrap sm:ml-3 hover:bg-blue-700 focus:outline-none focus:ring-2 focus:ring-offset-2 focus:ring-indigo-500",
    onClick: function onClick() {
      return uploadFile({
        file: _extends({}, file, {
          name: fileName
        }),
        folderId: folderId,
        appId: appId,
        consumerId: consumerId,
        serviceId: serviceId,
        jwt: jwt
      });
    }
  }, isLoading ? 'Uploading...' : 'Save file')));
};

var Search = function Search(_ref) {
  var setSearchTerm = _ref.setSearchTerm,
      searchTerm = _ref.searchTerm,
      isSearchVisible = _ref.isSearchVisible,
      setIsSearchVisible = _ref.setIsSearchVisible;
  var inputEl = useRef(null);

  var handleClick = function handleClick() {
    if (isSearchVisible) {
      setSearchTerm('');
      setIsSearchVisible(false);
    } else {
      setIsSearchVisible(true);
      setTimeout(function () {
        var _inputEl$current;

        (_inputEl$current = inputEl.current) == null ? void 0 : _inputEl$current.focus();
      }, 0);
    }
  };

  return isSearchVisible ? React.createElement("div", {
    className: "h-5"
  }, React.createElement("div", {
    className: "absolute w-full -inset-x-0 -top-2"
  }, React.createElement("div", {
    className: "relative"
  }, React.createElement("button", {
    className: "absolute right-2 top-2",
    onClick: function onClick() {
      return handleClick();
    }
  }, React.createElement("svg", {
    xmlns: "http://www.w3.org/2000/svg",
    className: "w-5 h-5 text-gray-400 transition duration-150 hover:text-gray-500",
    fill: "none",
    viewBox: "0 0 24 24",
    stroke: "currentColor"
  }, React.createElement("path", {
    strokeLinecap: "round",
    strokeLinejoin: "round",
    strokeWidth: 2,
    d: "M6 18L18 6M6 6l12 12"
  }))), React.createElement("input", {
    className: "w-full px-3 py-2 mb-3 text-sm leading-tight text-gray-700 border border-gray-200 rounded shadow-sm appearance-none focus:outline-none focus:shadow-outline",
    name: "search",
    placeholder: "Search",
    autoComplete: "off",
    ref: inputEl,
    onChange: function onChange(event) {
      return setSearchTerm(event.target.value);
    },
    value: searchTerm
  })))) : React.createElement("button", {
    className: "",
    onClick: function onClick() {
      return handleClick();
    }
  }, React.createElement("svg", {
    className: "w-5 h-5 text-gray-400 transition duration-150 hover:text-gray-500",
    xmlns: "http://www.w3.org/2000/svg",
    viewBox: "0 0 20 20",
    fill: "currentColor",
    "aria-hidden": "true"
  }, React.createElement("path", {
    fillRule: "evenodd",
    d: "M8 4a4 4 0 100 8 4 4 0 000-8zM2 8a6 6 0 1110.89 3.476l4.817 4.817a1 1 0 01-1.414 1.414l-4.816-4.816A6 6 0 012 8z",
    clipRule: "evenodd"
  })));
};

var SlideOver = function SlideOver(_ref) {
  var _window;

  var open = _ref.open,
      children = _ref.children;
  var desktopStyles = {
    width: 'calc(100% - 3rem)',
    left: '1.5rem'
  };
  var mobileStyles = {
    width: 'calc(100% - 2rem)',
    left: '1rem'
  };
  return React.createElement(Transition, {
    show: open,
    as: Fragment
  }, React.createElement("div", {
    className: "absolute bottom-0 left-0 right-0 rounded-t-2xl",
    style: ((_window = window) == null ? void 0 : _window.innerWidth) > 768 ? desktopStyles : mobileStyles
  }, React.createElement(Transition.Child, {
    as: Fragment,
    enter: "transform transition ease-in-out duration-300",
    enterFrom: "translate-y-full",
    enterTo: "translate-y-0",
    leave: "transform transition ease-in-out duration-300",
    leaveFrom: "translate-y-0",
    leaveTo: "translate-y-full"
  }, React.createElement("div", {
    className: "relative w-full rounded-t-2xl"
  }, React.createElement("div", {
    className: "flex flex-col h-full py-6 overflow-y-auto border border-gray-200 rounded-t-lg shadow-sm bg-gray-50"
  }, React.createElement("div", {
    className: "relative flex-1 px-4 sm:px-6"
  }, children))))));
};

var Spinner = function Spinner(_ref) {
  var _ref$className = _ref.className,
      className = _ref$className === void 0 ? 'w-8 h-8 text-blue-600' : _ref$className;
  return React.createElement("svg", {
    className: "animate-spin " + className,
    xmlns: "http://www.w3.org/2000/svg",
    fill: "none",
    viewBox: "0 0 24 24"
  }, React.createElement("circle", {
    className: "opacity-25",
    cx: "12",
    cy: "12",
    r: "10",
    stroke: "currentColor",
    strokeWidth: "4"
  }), React.createElement("path", {
    className: "opacity-75",
    fill: "currentColor",
    d: "M4 12a8 8 0 018-8V0C5.373 0 0 5.373 0 12h4zm2 5.291A7.962 7.962 0 014 12H0c0 3.042 1.135 5.824 3 7.938l3-2.647z"
  }));
};

var UploadButton = function UploadButton(_ref) {
  var folderId = _ref.folderId,
      appId = _ref.appId,
      consumerId = _ref.consumerId,
      serviceId = _ref.serviceId,
      jwt = _ref.jwt,
      onSuccess = _ref.onSuccess;

  var _useUpload = useUpload({
    onSuccess: onSuccess
  }),
      uploadFile = _useUpload.uploadFile,
      isLoading = _useUpload.isLoading;

  return React.createElement("div", {
    className: ""
  }, React.createElement("label", {
    htmlFor: "file-upload",
    className: "cursor-pointer"
  }, isLoading ? React.createElement(Spinner, {
    className: "w-5 h-5 text-gray-400"
  }) : React.createElement("svg", {
    xmlns: "http://www.w3.org/2000/svg",
    className: "w-5 h-5 text-gray-400 transition duration-150 hover:text-gray-500",
    fill: "none",
    viewBox: "0 0 24 24",
    stroke: "currentColor"
  }, React.createElement("path", {
    strokeLinecap: "round",
    strokeLinejoin: "round",
    strokeWidth: 2,
    d: "M7 16a4 4 0 01-.88-7.903A5 5 0 1115.9 6L16 6a5 5 0 011 9.9M15 13l-3-3m0 0l-3 3m3-3v12"
  })), React.createElement("input", {
    id: "file-upload",
    name: "file-upload",
    type: "file",
    className: "sr-only",
    onChange: function onChange(e) {
      var _e$target;

      return uploadFile({
        file: (_e$target = e.target) == null ? void 0 : _e$target.files[0],
        folderId: folderId,
        appId: appId,
        consumerId: consumerId,
        serviceId: serviceId,
        jwt: jwt
      });
    }
  })));
};

var useDebounce = function useDebounce(value, delay) {
  if (delay === void 0) {
    delay = 500;
  }

  var _useState = useState(value),
      currentValue = _useState[0],
      setCurrentValue = _useState[1];

  var interval = useRef(null);

  var clean = function clean() {
    if (interval.current !== null) {
      clearInterval(interval.current);
    }
  };

  useEffect(function () {
    interval.current = setTimeout(function () {
      setCurrentValue(value);
    }, delay);
    return clean;
  }, [value, delay]);
  useEffect(function () {
    return clean;
  }, []);
  return currentValue;
};

var usePrevious = function usePrevious(value) {
  var ref = useRef();
  useEffect(function () {
    ref.current = value;
  }, [value]);
  return ref.current;
};

/*! *****************************************************************************
Copyright (c) Microsoft Corporation.

Permission to use, copy, modify, and/or distribute this software for any
purpose with or without fee is hereby granted.

THE SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL WARRANTIES WITH
REGARD TO THIS SOFTWARE INCLUDING ALL IMPLIED WARRANTIES OF MERCHANTABILITY
AND FITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR ANY SPECIAL, DIRECT,
INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES WHATSOEVER RESULTING FROM
LOSS OF USE, DATA OR PROFITS, WHETHER IN AN ACTION OF CONTRACT, NEGLIGENCE OR
OTHER TORTIOUS ACTION, ARISING OUT OF OR IN CONNECTION WITH THE USE OR
PERFORMANCE OF THIS SOFTWARE.
***************************************************************************** */

function __awaiter(thisArg, _arguments, P, generator) {
    function adopt(value) { return value instanceof P ? value : new P(function (resolve) { resolve(value); }); }
    return new (P || (P = Promise))(function (resolve, reject) {
        function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
        function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
        function step(result) { result.done ? resolve(result.value) : adopt(result.value).then(fulfilled, rejected); }
        step((generator = generator.apply(thisArg, _arguments || [])).next());
    });
}

function __generator(thisArg, body) {
    var _ = { label: 0, sent: function() { if (t[0] & 1) throw t[1]; return t[1]; }, trys: [], ops: [] }, f, y, t, g;
    return g = { next: verb(0), "throw": verb(1), "return": verb(2) }, typeof Symbol === "function" && (g[Symbol.iterator] = function() { return this; }), g;
    function verb(n) { return function (v) { return step([n, v]); }; }
    function step(op) {
        if (f) throw new TypeError("Generator is already executing.");
        while (_) try {
            if (f = 1, y && (t = op[0] & 2 ? y["return"] : op[0] ? y["throw"] || ((t = y["return"]) && t.call(y), 0) : y.next) && !(t = t.call(y, op[1])).done) return t;
            if (y = 0, t) op = [op[0] & 2, t.value];
            switch (op[0]) {
                case 0: case 1: t = op; break;
                case 4: _.label++; return { value: op[1], done: false };
                case 5: _.label++; y = op[1]; op = [0]; continue;
                case 7: op = _.ops.pop(); _.trys.pop(); continue;
                default:
                    if (!(t = _.trys, t = t.length > 0 && t[t.length - 1]) && (op[0] === 6 || op[0] === 2)) { _ = 0; continue; }
                    if (op[0] === 3 && (!t || (op[1] > t[0] && op[1] < t[3]))) { _.label = op[1]; break; }
                    if (op[0] === 6 && _.label < t[1]) { _.label = t[1]; t = op; break; }
                    if (t && _.label < t[2]) { _.label = t[2]; _.ops.push(op); break; }
                    if (t[2]) _.ops.pop();
                    _.trys.pop(); continue;
            }
            op = body.call(thisArg, _);
        } catch (e) { op = [6, e]; y = 0; } finally { f = t = 0; }
        if (op[0] & 5) throw op[1]; return { value: op[0] ? op[1] : void 0, done: true };
    }
}

var noop = function () { };
// Using noop() as the undefined value as undefined can possibly be replaced
// by something else.  Prettier ignore and extra parentheses are necessary here
// to ensure that tsc doesn't remove the __NOINLINE__ comment.
// prettier-ignore
var UNDEFINED = ( /*#__NOINLINE__*/noop());
var OBJECT = Object;
var isUndefined = function (v) { return v === UNDEFINED; };
var isFunction = function (v) { return typeof v == 'function'; };
var STR_UNDEFINED = 'undefined';
var hasWindow = function () { return typeof window != STR_UNDEFINED; };

var IS_SERVER = !hasWindow() || 'Deno' in window;
// React currently throws a warning when using useLayoutEffect on the server.
// To get around it, we can conditionally useEffect on the server (no-op) and
// useLayoutEffect in the browser.
var useIsomorphicLayoutEffect = IS_SERVER ? useEffect : useLayoutEffect;

// use WeakMap to store the object->key mapping
// so the objects can be garbage collected.
// WeakMap uses a hashtable under the hood, so the lookup
// complexity is almost O(1).
var table = new WeakMap();
// counter of the key
var counter = 0;
// A stable hash implementation that supports:
// - Fast and ensures unique hash properties
// - Handles unserializable values
// - Handles object key ordering
// - Generates short results
//
// This is not a serialization function, and the result is not guaranteed to be
// parsible.
var stableHash = function (arg) {
    var type = typeof arg;
    var constructor = arg && arg.constructor;
    var isDate = constructor == Date;
    var result;
    var index;
    if (OBJECT(arg) === arg && !isDate && constructor != RegExp) {
        // Object/function, not null/date/regexp. Use WeakMap to store the id first.
        // If it's already hashed, directly return the result.
        result = table.get(arg);
        if (result)
            return result;
        // Store the hash first for circular reference detection before entering the
        // recursive `stableHash` calls.
        // For other objects like set and map, we use this id directly as the hash.
        result = ++counter + '~';
        table.set(arg, result);
        if (constructor == Array) {
            // Array.
            result = '@';
            for (index = 0; index < arg.length; index++) {
                result += stableHash(arg[index]) + ',';
            }
            table.set(arg, result);
        }
        if (constructor == OBJECT) {
            // Object, sort keys.
            result = '#';
            var keys = OBJECT.keys(arg).sort();
            while (!isUndefined((index = keys.pop()))) {
                if (!isUndefined(arg[index])) {
                    result += index + ':' + stableHash(arg[index]) + ',';
                }
            }
            table.set(arg, result);
        }
    }
    else {
        result = isDate
            ? arg.toJSON()
            : type == 'symbol'
                ? arg.toString()
                : type == 'string'
                    ? JSON.stringify(arg)
                    : '' + arg;
    }
    return result;
};

var serialize = function (key) {
    if (isFunction(key)) {
        try {
            key = key();
        }
        catch (err) {
            // dependencies not ready
            key = '';
        }
    }
    var args = [].concat(key);
    // If key is not falsy, or not an empty array, hash it.
    key =
        typeof key == 'string'
            ? key
            : (Array.isArray(key)
                ? key.length
                : key)
                ? stableHash(key)
                : '';
    var errorKey = key ? '$err$' + key : '';
    var isValidatingKey = key ? '$req$' + key : '';
    return [key, args, errorKey, isValidatingKey];
};

/*! *****************************************************************************
Copyright (c) Microsoft Corporation.

Permission to use, copy, modify, and/or distribute this software for any
purpose with or without fee is hereby granted.

THE SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL WARRANTIES WITH
REGARD TO THIS SOFTWARE INCLUDING ALL IMPLIED WARRANTIES OF MERCHANTABILITY
AND FITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR ANY SPECIAL, DIRECT,
INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES WHATSOEVER RESULTING FROM
LOSS OF USE, DATA OR PROFITS, WHETHER IN AN ACTION OF CONTRACT, NEGLIGENCE OR
OTHER TORTIOUS ACTION, ARISING OUT OF OR IN CONNECTION WITH THE USE OR
PERFORMANCE OF THIS SOFTWARE.
***************************************************************************** */

var __assign = function() {
    __assign = Object.assign || function __assign(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p)) t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};

var normalize = function (args) {
    return isFunction(args[1])
        ? [args[0], args[1], args[2] || {}]
        : [args[0], null, (args[1] === null ? args[2] : args[1]) || {}];
};

// Create a custom hook with a middleware
var withMiddleware = function (useSWR, middleware) {
    return function () {
        var args = [];
        for (var _i = 0; _i < arguments.length; _i++) {
            args[_i] = arguments[_i];
        }
        var _a = normalize(args), key = _a[0], fn = _a[1], config = _a[2];
        var middlewares = (config.use || []).concat(middleware);
        return useSWR(key, fn, __assign(__assign({}, config), { use: middlewares }));
    };
};

// We have to several type castings here because `useSWRInfinite` is a special
var INFINITE_PREFIX = '$inf$';
var getFirstPageKey = function (getKey) {
    return serialize(getKey ? getKey(0, null) : null)[0];
};
var infinite = (function (useSWRNext) {
    return function (getKey, fn, config) {
        var rerender = useState({})[1];
        var didMountRef = useRef(false);
        var dataRef = useRef();
        var cache = config.cache, _a = config.initialSize, initialSize = _a === void 0 ? 1 : _a, _b = config.revalidateAll, revalidateAll = _b === void 0 ? false : _b, _c = config.persistSize, persistSize = _c === void 0 ? false : _c, _d = config.revalidateFirstPage, revalidateFirstPage = _d === void 0 ? true : _d;
        // The serialized key of the first page.
        var firstPageKey = null;
        try {
            firstPageKey = getFirstPageKey(getKey);
        }
        catch (err) {
            // not ready
        }
        // We use cache to pass extra info (context) to fetcher so it can be globally
        // shared. The key of the context data is based on the first page key.
        var contextCacheKey = null;
        // Page size is also cached to share the page data between hooks with the
        // same key.
        var pageSizeCacheKey = null;
        if (firstPageKey) {
            contextCacheKey = '$ctx$' + firstPageKey;
            pageSizeCacheKey = '$len$' + firstPageKey;
        }
        var resolvePageSize = useCallback(function () {
            var cachedPageSize = cache.get(pageSizeCacheKey);
            return isUndefined(cachedPageSize) ? initialSize : cachedPageSize;
            // `cache` isn't allowed to change during the lifecycle
            // eslint-disable-next-line react-hooks/exhaustive-deps
        }, [pageSizeCacheKey, initialSize]);
        // keep the last page size to restore it with the persistSize option
        var lastPageSizeRef = useRef(resolvePageSize());
        // When the page key changes, we reset the page size if it's not persisted
        useIsomorphicLayoutEffect(function () {
            if (!didMountRef.current) {
                didMountRef.current = true;
                return;
            }
            if (firstPageKey) {
                // If the key has been changed, we keep the current page size if persistSize is enabled
                cache.set(pageSizeCacheKey, persistSize ? lastPageSizeRef.current : initialSize);
            }
            // `initialSize` isn't allowed to change during the lifecycle
            // eslint-disable-next-line react-hooks/exhaustive-deps
        }, [firstPageKey]);
        // Actual SWR hook to load all pages in one fetcher.
        var swr = useSWRNext(firstPageKey ? INFINITE_PREFIX + firstPageKey : null, function () { return __awaiter(void 0, void 0, void 0, function () {
            var _a, forceRevalidateAll, originalData, data, pageSize, previousPageData, i, _b, pageKey, pageArgs, pageData, shouldFetchPage;
            return __generator(this, function (_c) {
                switch (_c.label) {
                    case 0:
                        _a = cache.get(contextCacheKey) || [], forceRevalidateAll = _a[0], originalData = _a[1];
                        data = [];
                        pageSize = resolvePageSize();
                        previousPageData = null;
                        i = 0;
                        _c.label = 1;
                    case 1:
                        if (!(i < pageSize)) return [3 /*break*/, 5];
                        _b = serialize(getKey ? getKey(i, previousPageData) : null), pageKey = _b[0], pageArgs = _b[1];
                        if (!pageKey) {
                            // `pageKey` is falsy, stop fetching new pages.
                            return [3 /*break*/, 5];
                        }
                        pageData = cache.get(pageKey);
                        shouldFetchPage = revalidateAll ||
                            forceRevalidateAll ||
                            isUndefined(pageData) ||
                            (revalidateFirstPage && !i && !isUndefined(dataRef.current)) ||
                            (originalData &&
                                !isUndefined(originalData[i]) &&
                                !config.compare(originalData[i], pageData));
                        if (!(fn && shouldFetchPage)) return [3 /*break*/, 3];
                        return [4 /*yield*/, fn.apply(void 0, pageArgs)];
                    case 2:
                        pageData = _c.sent();
                        cache.set(pageKey, pageData);
                        _c.label = 3;
                    case 3:
                        data.push(pageData);
                        previousPageData = pageData;
                        _c.label = 4;
                    case 4:
                        ++i;
                        return [3 /*break*/, 1];
                    case 5:
                        // once we executed the data fetching based on the context, clear the context
                        cache.delete(contextCacheKey);
                        // return the data
                        return [2 /*return*/, data];
                }
            });
        }); }, config);
        // update dataRef
        useIsomorphicLayoutEffect(function () {
            dataRef.current = swr.data;
        }, [swr.data]);
        var mutate = useCallback(function () {
            var args = [];
            for (var _i = 0; _i < arguments.length; _i++) {
                args[_i] = arguments[_i];
            }
            var data = args[0];
            // Default to true.
            var shouldRevalidate = args[1] !== false;
            // It is possible that the key is still falsy.
            if (!contextCacheKey)
                return;
            if (shouldRevalidate) {
                if (!isUndefined(data)) {
                    // We only revalidate the pages that are changed
                    var originalData = dataRef.current;
                    cache.set(contextCacheKey, [false, originalData]);
                }
                else {
                    // Calling `mutate()`, we revalidate all pages
                    cache.set(contextCacheKey, [true]);
                }
            }
            return args.length ? swr.mutate(data, shouldRevalidate) : swr.mutate();
        }, 
        // swr.mutate is always the same reference
        // eslint-disable-next-line react-hooks/exhaustive-deps
        [contextCacheKey]);
        // Function to load pages data from the cache based on the page size.
        var resolvePagesFromCache = function (pageSize) {
            // return an array of page data
            var data = [];
            var previousPageData = null;
            for (var i = 0; i < pageSize; ++i) {
                var pageKey = serialize(getKey ? getKey(i, previousPageData) : null)[0];
                // Get the cached page data.
                var pageData = pageKey ? cache.get(pageKey) : UNDEFINED;
                // Return the current data if we can't get it from the cache.
                if (isUndefined(pageData))
                    return dataRef.current;
                data.push(pageData);
                previousPageData = pageData;
            }
            // Return the data
            return data;
        };
        // Extend the SWR API
        var setSize = useCallback(function (arg) {
            // It is possible that the key is still falsy.
            if (!pageSizeCacheKey)
                return;
            var size;
            if (isFunction(arg)) {
                size = arg(resolvePageSize());
            }
            else if (typeof arg == 'number') {
                size = arg;
            }
            if (typeof size != 'number')
                return;
            cache.set(pageSizeCacheKey, size);
            lastPageSizeRef.current = size;
            rerender({});
            return mutate(resolvePagesFromCache(size));
        }, 
        // `cache` and `rerender` isn't allowed to change during the lifecycle
        // eslint-disable-next-line react-hooks/exhaustive-deps
        [pageSizeCacheKey, resolvePageSize, mutate]);
        // Use getter functions to avoid unnecessary re-renders caused by triggering
        // all the getters of the returned swr object.
        return {
            size: resolvePageSize(),
            setSize: setSize,
            mutate: mutate,
            get error() {
                return swr.error;
            },
            get data() {
                return swr.data;
            },
            get isValidating() {
                return swr.isValidating;
            }
        };
    };
});
var index = withMiddleware(useSWR, infinite);

var FilesContainer = function FilesContainer(_ref) {
  var _data2, _files2, _folders, _files3;

  var appId = _ref.appId,
      consumerId = _ref.consumerId,
      jwt = _ref.jwt,
      onSelect = _ref.onSelect,
      connections = _ref.connections,
      connection = _ref.connection,
      setConnection = _ref.setConnection,
      fileToSave = _ref.fileToSave;

  var _useState = useState(null),
      folderId = _useState[0],
      setFolderId = _useState[1];

  var _useState2 = useState([]),
      folders = _useState2[0],
      setFolders = _useState2[1];

  var _useState3 = useState(null),
      file = _useState3[0],
      setFile = _useState3[1];

  var _useState4 = useState(''),
      searchTerm = _useState4[0],
      setSearchTerm = _useState4[1];

  var _useState5 = useState(),
      searchResults = _useState5[0],
      setSearchResults = _useState5[1];

  var _useState6 = useState(false),
      isSearching = _useState6[0],
      setIsSearching = _useState6[1];

  var _useState7 = useState(false),
      isSearchVisible = _useState7[0],
      setIsSearchVisible = _useState7[1];

  var debouncedSearchTerm = useDebounce(searchTerm);
  var serviceId = connection.service_id;
  var prevServiceId = usePrevious(serviceId);
  var prevFolderId = usePrevious(folderId);
  var searchMode = !!(debouncedSearchTerm != null && debouncedSearchTerm.length);
  var headers = {
    'Content-Type': 'application/json',
    'x-apideck-auth-type': 'JWT',
    'x-apideck-app-id': appId,
    'x-apideck-consumer-id': consumerId,
    'x-apideck-service-id': serviceId,
    Authorization: "Bearer " + jwt
  };

  var fetcher = /*#__PURE__*/function () {
    var _ref2 = _asyncToGenerator( /*#__PURE__*/runtime_1.mark(function _callee(url) {
      var response;
      return runtime_1.wrap(function _callee$(_context) {
        while (1) {
          switch (_context.prev = _context.next) {
            case 0:
              _context.next = 2;
              return fetch(url, {
                headers: headers
              });

            case 2:
              response = _context.sent;
              _context.next = 5;
              return response.json();

            case 5:
              return _context.abrupt("return", _context.sent);

            case 6:
            case "end":
              return _context.stop();
          }
        }
      }, _callee);
    }));

    return function fetcher(_x) {
      return _ref2.apply(this, arguments);
    };
  }();

  var getKey = function getKey(pageIndex, previousPage) {
    var _previousPage$data, _previousPage$meta, _previousPage$meta$cu;

    // If we switch from connector we want the folder ID to always be root
    var id = prevServiceId && prevServiceId !== serviceId || !folderId ? 'root' : folderId;
    var filterParams = id === 'shared' ? 'filter[shared]=true' : "filter[folder_id]=" + id;
    var fileUrl = "https://unify.apideck.com/file-storage/files?limit=30&" + filterParams;
    if (previousPage && !(previousPage != null && (_previousPage$data = previousPage.data) != null && _previousPage$data.length)) return null;
    if (pageIndex === 0) return fileUrl + "#serviceId=" + serviceId;
    var cursor = previousPage == null ? void 0 : (_previousPage$meta = previousPage.meta) == null ? void 0 : (_previousPage$meta$cu = _previousPage$meta.cursors) == null ? void 0 : _previousPage$meta$cu.next; // We add the serviceId to the end of the URL so SWR caches the request results per service

    return fileUrl + "&cursor=" + cursor + "#serviceId=" + serviceId;
  };

  var _useSWRInfinite = index(getKey, fetcher, {
    shouldRetryOnError: false
  }),
      data = _useSWRInfinite.data,
      setSize = _useSWRInfinite.setSize,
      size = _useSWRInfinite.size,
      error = _useSWRInfinite.error,
      mutate = _useSWRInfinite.mutate;

  var _useUpload = useUpload({
    onSuccess: function onSuccess() {
      mutate();
      hideDropzone();
    }
  }),
      uploadFile = _useUpload.uploadFile,
      isUploading = _useUpload.isLoading,
      progress = _useUpload.progress;

  var onDrop = /*#__PURE__*/function () {
    var _ref3 = _asyncToGenerator( /*#__PURE__*/runtime_1.mark(function _callee2(acceptedFiles) {
      return runtime_1.wrap(function _callee2$(_context2) {
        while (1) {
          switch (_context2.prev = _context2.next) {
            case 0:
              _context2.next = 2;
              return uploadFile({
                file: acceptedFiles[0],
                folderId: folderId,
                appId: appId,
                consumerId: consumerId,
                serviceId: serviceId,
                jwt: jwt
              });

            case 2:
              hideDropzone();

            case 3:
            case "end":
              return _context2.stop();
          }
        }
      }, _callee2);
    }));

    return function onDrop(_x2) {
      return _ref3.apply(this, arguments);
    };
  }();

  var _useDropzone = useDropzone({
    onDrop: onDrop
  }),
      getRootProps = _useDropzone.getRootProps,
      getInputProps = _useDropzone.getInputProps,
      isDragActive = _useDropzone.isDragActive;

  var nextPage = function nextPage() {
    var _data, _data$meta, _data$meta$cursors;

    var nextCursor = (data == null ? void 0 : data.length) && ((_data = data[data.length - 1]) == null ? void 0 : (_data$meta = _data.meta) == null ? void 0 : (_data$meta$cursors = _data$meta.cursors) == null ? void 0 : _data$meta$cursors.next);

    if (nextCursor) {
      setSize(size + 1);
    }
  };

  var hideDropzone = function hideDropzone() {
    var _dropzoneElement$styl, _dropzoneElement$styl2;

    var dropzoneElement = document.querySelector('.dropzone');
    if (dropzoneElement != null && (_dropzoneElement$styl = dropzoneElement.style) != null && _dropzoneElement$styl.visibility) dropzoneElement.style.visibility = 'hidden';
    if (dropzoneElement != null && (_dropzoneElement$styl2 = dropzoneElement.style) != null && _dropzoneElement$styl2.opacity) dropzoneElement.style.opacity = '0';
  };

  var showDropzone = function showDropzone() {
    var dropzoneElement = document.querySelector('.dropzone');
    if (dropzoneElement != null && dropzoneElement.style) dropzoneElement.style.visibility = 'visible';
    if (dropzoneElement != null && dropzoneElement.style) dropzoneElement.style.opacity = '1';
  };

  useEffect(function () {
    var lastTarget = null;

    var onEnter = function onEnter(e) {
      lastTarget = e.target; // cache the last target here

      showDropzone();
    };

    var onLeave = function onLeave(e) {
      if (e.target === lastTarget || e.target === document) {
        hideDropzone();
      }
    };

    var addDragListeners = function addDragListeners() {
      window.addEventListener('dragenter', onEnter);
      window.addEventListener('dragleave', onLeave);
    };

    var cleanupListeners = function cleanupListeners() {
      window.removeEventListener('dragenter', onEnter);
      window.removeEventListener('dragleave', onLeave);
    };

    addDragListeners();
    return function () {
      return cleanupListeners();
    };
  }, []);
  useEffect(function () {
    // If we switch from connector we want the folder ID to always be root except when we are in search mode
    if (prevServiceId && prevServiceId !== serviceId && prevFolderId === folderId || !folderId) {
      setFolderId('root');
      setFolders([]);
    }
  }, [serviceId, prevServiceId]);

  var handleSelect = function handleSelect(file) {
    if (file.type === 'folder') {
      setFolderId(file.id);

      if (searchMode) {
        var _file$connection;

        setSearchTerm('');
        setIsSearchVisible(false);
        setFolders([file]); // When clicking a folder in search mode, we want to change the selected connection if the folder is from a different service

        if (((_file$connection = file.connection) == null ? void 0 : _file$connection.service_id) !== serviceId) {
          setConnection(file.connection);
        }
      } else {
        setFolders([].concat(folders, [file]));
      }
    }

    if (file.type === 'file') {
      setFile(file);
    }
  };

  var handleBreadcrumbClick = function handleBreadcrumbClick(file) {
    if (file) {
      setFolderId(file.id);
      var index = folders.indexOf(file);
      var newArray = [].concat(folders.slice(0, index + 1));
      setFolders(newArray);
    } else {
      setFolderId('root');
      setFolders([]);
    }
  };

  var filesError = error || (data == null ? void 0 : data.length) && ((_data2 = data[data.length - 1]) == null ? void 0 : _data2.error);
  var isLoading = !data && !error;
  var isLoadingMore = size > 0 && data && typeof data[size - 1] === 'undefined';
  var files = data != null && data.length ? data.map(function (page) {
    return page == null ? void 0 : page.data;
  }).flat() : data ? [data] : []; // Add Google Drive shared folder to root

  if ((!folderId || folderId === 'root') && data != null && data.length && serviceId === 'google-drive') {
    var _files;

    var sharedFolder = {
      id: 'shared',
      name: 'Shared with me',
      type: 'folder'
    };

    if ((_files = files) != null && _files.length) {
      files = [sharedFolder].concat(files);
    } else {
      files.push(sharedFolder);
    }
  }

  useEffect(function () {
    if (debouncedSearchTerm != null && debouncedSearchTerm.length) {
      setIsSearching(true);

      var searchFiles = /*#__PURE__*/function () {
        var _ref4 = _asyncToGenerator( /*#__PURE__*/runtime_1.mark(function _callee3(serviceId) {
          var url, options, response;
          return runtime_1.wrap(function _callee3$(_context3) {
            while (1) {
              switch (_context3.prev = _context3.next) {
                case 0:
                  url = 'https://unify.apideck.com/file-storage/files/search';
                  options = {
                    headers: _extends({}, headers, {
                      'x-apideck-service-id': serviceId
                    }),
                    method: 'POST',
                    body: JSON.stringify({
                      query: debouncedSearchTerm
                    })
                  };
                  _context3.next = 4;
                  return fetch(url, options);

                case 4:
                  response = _context3.sent;
                  return _context3.abrupt("return", response.json());

                case 6:
                case "end":
                  return _context3.stop();
              }
            }
          }, _callee3);
        }));

        return function searchFiles(_x3) {
          return _ref4.apply(this, arguments);
        };
      }(); // Search across all callable connections


      var promises = connections.map(function (con) {
        return searchFiles(con.service_id);
      });
      Promise.all(promises).then(function (responses) {
        var results = responses == null ? void 0 : responses.map(function (res) {
          var _res$data;

          return (_res$data = res.data) == null ? void 0 : _res$data.map(function (file) {
            return _extends({}, file, {
              connection: connections.find(function (con) {
                return con.service_id === res.service;
              })
            });
          });
        }).flat(); // Make sure the search results of the currently active connection are first

        var currentConnectionResults = results.filter(function (file) {
          var _file$connection2;

          return ((_file$connection2 = file.connection) == null ? void 0 : _file$connection2.service_id) === connection.service_id;
        });
        var rest = results.filter(function (file) {
          var _file$connection3;

          return ((_file$connection3 = file.connection) == null ? void 0 : _file$connection3.service_id) !== connection.service_id;
        });
        setSearchResults([].concat(currentConnectionResults, rest));
      })["finally"](function () {
        return setIsSearching(false);
      });
    }
  }, [debouncedSearchTerm, serviceId]);
  var hasFiles = searchMode ? searchResults == null ? void 0 : searchResults.length : (_files2 = files) == null ? void 0 : _files2.length;
  return React.createElement(Fragment, null, React.createElement("div", {
    className: "relative flex items-center justify-between mb-2"
  }, React.createElement(Breadcrumbs, {
    folders: folders,
    handleClick: handleBreadcrumbClick
  }), React.createElement("div", {
    className: "flex items-center space-x-2"
  }, React.createElement(Search, {
    searchTerm: searchTerm,
    setSearchTerm: setSearchTerm,
    isSearchVisible: isSearchVisible,
    setIsSearchVisible: setIsSearchVisible
  }), !fileToSave ? React.createElement(UploadButton, {
    file: fileToSave,
    consumerId: consumerId,
    jwt: jwt,
    folderId: folderId,
    serviceId: serviceId,
    appId: appId,
    onSuccess: mutate
  }) : null)), isLoading || isSearching ? React.createElement(LoadingTable, {
    isSearching: isSearching
  }) : null, React.createElement("div", Object.assign({}, getRootProps(), {
    className: "relative"
  }), React.createElement("input", Object.assign({}, getInputProps())), isDragActive || isUploading ? React.createElement("div", {
    className: "flex justify-center px-6 py-24 mt-4 border-2 border-gray-300 border-dashed rounded-md"
  }, isUploading ? React.createElement("div", {
    className: "flex flex-col items-center"
  }, React.createElement(Spinner, {
    className: "w-6 h-6 text-gray-500"
  }), progress !== null ? React.createElement("div", {
    className: "w-full h-2 mt-4 bg-gray-200 rounded",
    style: {
      width: 200
    }
  }, React.createElement("div", {
    className: "h-2 bg-blue-400 rounded " + (progress === 100 && 'animate-ping') + " ",
    style: {
      width: progress + "%"
    }
  })) : null) : React.createElement("div", {
    className: "space-y-1 text-center"
  }, React.createElement("svg", {
    className: "w-12 h-12 mx-auto text-gray-400",
    stroke: "currentColor",
    fill: "none",
    viewBox: "0 0 48 48",
    "aria-hidden": "true"
  }, React.createElement("path", {
    d: "M28 8H12a4 4 0 00-4 4v20m32-12v8m0 0v8a4 4 0 01-4 4H12a4 4 0 01-4-4v-4m32-4l-3.172-3.172a4 4 0 00-5.656 0L28 28M8 32l9.172-9.172a4 4 0 015.656 0L28 28m0 0l4 4m4-24h8m-4-4v8m-12 4h.02",
    strokeWidth: 2,
    strokeLinecap: "round",
    strokeLinejoin: "round"
  })), React.createElement("div", {
    className: "flex text-sm text-gray-600"
  }, React.createElement("label", {
    htmlFor: "file-upload",
    className: "relative bg-white rounded-md"
  }, React.createElement("span", null, "Drop here to upload to"), React.createElement("input", {
    id: "file-upload",
    name: "file-upload",
    type: "file",
    className: "sr-only"
  })), React.createElement("p", {
    className: "pl-1 font-medium text-indigo-600"
  }, connection.name)), React.createElement("p", {
    className: "text-xs text-gray-500"
  }, folders.length ? "Folder: " + ((_folders = folders[folders.length - 1]) == null ? void 0 : _folders.name) : 'Root folder'))) : React.createElement("div", {
    className: "absolute inset-0 invisible w-full h-96 dropzone"
  })), !isLoading && !isSearching && hasFiles && !filesError && !isDragActive && !isUploading ? React.createElement(FilesTable, {
    data: searchMode && searchResults ? searchResults : files,
    handleSelect: handleSelect,
    isLoadingMore: isLoadingMore,
    searchMode: searchMode,
    uploadingMode: !!fileToSave
  }) : null, !isLoading && !isSearching && !hasFiles ? React.createElement("p", {
    className: "py-4 text-sm text-center text-gray-700"
  }, "No files found") : null, !isLoading ? React.createElement(SlideOver, {
    open: !!file
  }, React.createElement(FileDetails, {
    file: file,
    setFile: setFile,
    onSelect: onSelect
  })) : null, !isLoading && filesError ? React.createElement("p", {
    className: "mt-2 text-sm text-red-600"
  }, (filesError == null ? void 0 : filesError.message) || filesError) : null, (_files3 = files) != null && _files3.length && !isLoadingMore && !searchMode ? React.createElement("div", {
    className: "flex flex-row-reverse py-4 border-gray-200"
  }, React.createElement(Waypoint, {
    onEnter: function onEnter() {
      return nextPage();
    }
  })) : null, fileToSave && React.createElement(SaveFileForm, {
    file: fileToSave,
    appId: appId,
    consumerId: consumerId,
    jwt: jwt,
    folderId: folderId,
    serviceId: serviceId,
    onSuccess: onSelect
  }));
};

var SelectConnection = function SelectConnection(_ref) {
  var jwt = _ref.jwt,
      connections = _ref.connections,
      connection = _ref.connection,
      setConnection = _ref.setConnection,
      isLoading = _ref.isLoading;

  var redirectToVault = function redirectToVault() {
    var redirectUrl = "https://vault.apideck.com/session/" + jwt;
    window.location.href = redirectUrl;
  };

  var statusColor = function statusColor(connection) {
    if (!connection.enabled) return 'bg-gray-300';
    if (connection.state !== 'callable') return 'bg-yellow-400';
    return 'bg-green-400';
  };

  var handleClick = function handleClick(connection) {
    if (connection.state === 'callable') {
      setConnection(connection);
      return;
    }

    if (connection.state === 'available') {
      // Enable integration in vault
      var _redirectUrl = "https://vault.apideck.com/integrations/file-storage/" + (connection == null ? void 0 : connection.service_id) + "/enable?jwt=" + jwt;

      window.location.href = _redirectUrl;
      return;
    } // Redirect to integration settings page


    var redirectUrl = "https://vault.apideck.com/integrations/file-storage/" + (connection == null ? void 0 : connection.service_id) + "?jwt=" + jwt;
    window.location.href = redirectUrl;
  };

  return React.createElement("div", {
    className: "relative inline-block"
  }, React.createElement(Menu, null, function (_ref2) {
    var open = _ref2.open;
    return React.createElement(React.Fragment, null, React.createElement(Menu.Button, {
      className: "flex items-center justify-between w-full px-4 py-2 text-sm font-medium text-blue-800 bg-blue-100 border border-blue-200 rounded-md group hover:bg-cool-gray-200 focus:outline-none focus:ring-2 focus:ring-offset-2 focus:ring-offset-cool-gray-100 focus:ring-blue-600",
      style: {
        minWidth: 180
      }
    }, React.createElement("div", null, !isLoading && (connection == null ? void 0 : connection.icon) && React.createElement("img", {
      className: "inline-block w-6 h-6 mr-2 rounded-full " + (isLoading ? 'animate-spin opacity-20' : ''),
      src: !isLoading && connection != null && connection.icon ? connection == null ? void 0 : connection.icon : '/img/logo.png',
      alt: ""
    }), isLoading && React.createElement(Spinner, {
      className: "w-6 h-6"
    }), !isLoading && React.createElement("span", null, (connection == null ? void 0 : connection.name) || 'No integrations')), React.createElement("svg", {
      className: "w-5 h-5 ml-2 -mr-1",
      viewBox: "0 0 20 20",
      fill: "currentColor"
    }, React.createElement("path", {
      fillRule: "evenodd",
      d: "M5.293 7.293a1 1 0 011.414 0L10 10.586l3.293-3.293a1 1 0 111.414 1.414l-4 4a1 1 0 01-1.414 0l-4-4a1 1 0 010-1.414z",
      clipRule: "evenodd"
    }))), React.createElement(Transition, {
      show: open,
      enter: "transition ease-out duration-100",
      enterFrom: "transform opacity-0 scale-95",
      enterTo: "transform opacity-100 scale-100",
      leave: "transition ease-in duration-75",
      leaveFrom: "transform opacity-100 scale-100",
      leaveTo: "transform opacity-0 scale-95",
      className: "min-w-sm"
    }, React.createElement(Menu.Items, {
      "static": true,
      className: "absolute right-0 z-10 w-full mt-2 origin-top-right bg-white border divide-y rounded-md outline-none border-cool-gray-200 divide-cool-gray-100"
    }, React.createElement("div", {
      className: "py-1"
    }, connections == null ? void 0 : connections.map(function (connection, i) {
      return React.createElement(Menu.Item, {
        key: i
      }, function (_ref3) {
        var active = _ref3.active;
        return React.createElement("div", {
          onClick: function onClick() {
            return handleClick(connection);
          },
          className: (active ? 'bg-gray-100 text-gray-900' : 'text-gray-600') + " flex items-center justify-between min-w-0 mx-2 cursor-pointer rounded-md py-0.5 overflow-hidden " + (connection.enabled ? '' : 'opacity-60')
        }, React.createElement("img", {
          className: "flex-shrink-0 w-6 h-6 m-2 rounded-full",
          src: connection.icon,
          alt: ""
        }), React.createElement("span", {
          className: "flex-1 min-w-0"
        }, React.createElement("span", {
          className: "text-sm font-medium text-gray-900 truncate"
        }, connection.name)), React.createElement("span", {
          className: "inline-block w-2.5 h-2.5 mr-2 rounded-full ring-2 ring-white " + statusColor(connection)
        }));
      });
    }), React.createElement(Menu.Item, null, function (_ref4) {
      var active = _ref4.active;
      return React.createElement("div", {
        onClick: function onClick() {
          return redirectToVault();
        },
        className: (active ? 'bg-gray-100 text-gray-900' : 'text-gray-600') + " flex items-center justify-between min-w-0 mx-2 cursor-pointer rounded-md py-0.5 overflow-hidden"
      }, React.createElement("svg", {
        className: "flex-shrink-0 w-6 h-6 m-2",
        xmlns: "http://www.w3.org/2000/svg",
        fill: "none",
        viewBox: "0 0 24 24",
        stroke: "currentColor"
      }, React.createElement("path", {
        strokeLinecap: "round",
        strokeLinejoin: "round",
        strokeWidth: 2,
        d: "M12 6v6m0 0v6m0-6h6m-6 0H6"
      })), React.createElement("span", {
        className: "flex-1 min-w-0"
      }, React.createElement("span", {
        className: "text-sm font-medium text-gray-900 truncate"
      }, "Add integration")));
    })))));
  }));
};

/**
 * The Apideck File Picker
 */

var ModalContent = function ModalContent(_ref) {
  var _document$getElementB;

  var appId = _ref.appId,
      consumerId = _ref.consumerId,
      jwt = _ref.jwt,
      onSelect = _ref.onSelect,
      onConnectionSelect = _ref.onConnectionSelect,
      title = _ref.title,
      subTitle = _ref.subTitle,
      fileToSave = _ref.fileToSave;

  var _useState = useState(),
      connection = _useState[0],
      setConnection = _useState[1];

  var getConnections = /*#__PURE__*/function () {
    var _ref2 = _asyncToGenerator( /*#__PURE__*/runtime_1.mark(function _callee(url) {
      var response;
      return runtime_1.wrap(function _callee$(_context) {
        while (1) {
          switch (_context.prev = _context.next) {
            case 0:
              _context.next = 2;
              return fetch(url, {
                headers: {
                  'Content-Type': 'application/json',
                  'x-apideck-consumer-id': consumerId,
                  'x-apideck-app-id': appId,
                  'x-apideck-auth-type': 'JWT',
                  Authorization: "Bearer " + jwt
                }
              });

            case 2:
              response = _context.sent;
              _context.next = 5;
              return response.json();

            case 5:
              return _context.abrupt("return", _context.sent);

            case 6:
            case "end":
              return _context.stop();
          }
        }
      }, _callee);
    }));

    return function getConnections(_x) {
      return _ref2.apply(this, arguments);
    };
  }();

  var _useSWR = useSWR("https://unify.apideck.com/vault/connections?api=file-storage", getConnections, {
    shouldRetryOnError: false
  }),
      data = _useSWR.data,
      error = _useSWR.error;

  var isLoading = !data && !error;
  var hasError = (data == null ? void 0 : data.error) || error;
  var connections = data == null ? void 0 : data.data;
  var callableConnections = connections == null ? void 0 : connections.filter(function (connection) {
    return connection.state === 'callable';
  });
  useEffect(function () {
    if (!connection && callableConnections != null && callableConnections.length) {
      setConnection(callableConnections[0]);
    } // eslint-disable-next-line react-hooks/exhaustive-deps

  }, [setConnection, callableConnections]);
  useEffect(function () {
    if (connection) {
      onConnectionSelect(connection);
    }
  }, [connection]);
  var modalHeight = (_document$getElementB = document.getElementById('modal-component')) == null ? void 0 : _document$getElementB.clientHeight;
  return React.createElement("div", {
    className: "relative -m-6 bg-white sm:rounded-lg h-modal",
    style: {
      height: '34rem'
    }
  }, React.createElement("div", {
    className: "flex items-center justify-between px-4 py-5 sm:px-6"
  }, React.createElement("div", null, React.createElement("h3", {
    className: "text-lg font-medium leading-6 text-gray-900"
  }, title), React.createElement("p", {
    className: "max-w-2xl mt-1 text-sm text-gray-500"
  }, hasError ? React.createElement("span", {
    className: "mb-2 text-red-600"
  }, hasError) : React.createElement("span", {
    className: "text-gray-700 dark:text-gray-400"
  }, connection ? subTitle : 'No connector selected'))), React.createElement(SelectConnection, {
    jwt: jwt,
    connections: connections,
    connection: connection,
    setConnection: setConnection,
    isLoading: isLoading
  })), React.createElement("div", {
    className: "px-4 py-5 overflow-y-auto border-t border-gray-200 sm:px-6",
    style: {
      height: typeof window !== 'undefined' && modalHeight ? modalHeight - 70 : 'calc(100% - 70px)'
    }
  }, connection ? React.createElement(ToastProvider, null, React.createElement(FilesContainer, {
    appId: appId,
    consumerId: consumerId,
    jwt: jwt,
    onSelect: onSelect,
    connections: callableConnections,
    connection: connection,
    setConnection: setConnection,
    fileToSave: fileToSave
  })) : !(callableConnections != null && callableConnections.length) && !isLoading ? React.createElement("div", {
    className: "flex items-center justify-center border-2 border-gray-200 border-dashed rounded-lg h-96"
  }, React.createElement("div", {
    className: "text-center"
  }, React.createElement("a", {
    href: "https://vault.apideck.com/session/" + jwt,
    target: "_blank",
    rel: "noreferrer",
    className: "text-indigo-600 hover:text-indigo-900"
  }, "Go to vault"), ' ', "to add file storage connectors")) : null));
};

var _excluded$1 = ["connection"];
/**
 * The Apideck File Picker component
 */

var FilePicker = /*#__PURE__*/forwardRef(function FilePicker(_ref, ref) {
  var appId = _ref.appId,
      consumerId = _ref.consumerId,
      jwt = _ref.jwt,
      trigger = _ref.trigger,
      onSelect = _ref.onSelect,
      onConnectionSelect = _ref.onConnectionSelect,
      title = _ref.title,
      subTitle = _ref.subTitle,
      _ref$showAttribution = _ref.showAttribution,
      showAttribution = _ref$showAttribution === void 0 ? true : _ref$showAttribution,
      _ref$open = _ref.open,
      open = _ref$open === void 0 ? false : _ref$open,
      onClose = _ref.onClose,
      fileToSave = _ref.fileToSave;

  var _useState = useState(false),
      isOpen = _useState[0],
      setIsOpen = _useState[1];

  var handleFileSelect = function handleFileSelect(file) {
    var fileToReturn = file;

    if (file.connection) {
      var rest = _objectWithoutPropertiesLoose(file, _excluded$1);

      fileToReturn = rest;
    }

    if (onSelect) onSelect(fileToReturn);
    onCloseModal();
  };

  var handleConnectionSelect = function handleConnectionSelect(connection) {
    if (onConnectionSelect) onConnectionSelect(connection);
  };

  var onCloseModal = function onCloseModal() {
    setIsOpen(false);

    if (onClose) {
      onClose();
    }
  };

  useEffect(function () {
    if (open) {
      setIsOpen(true);
    }
  }, [open]);
  return React.createElement(Fragment, null, trigger ? React.cloneElement(trigger, {
    onClick: function onClick() {
      return setIsOpen(true);
    },
    ref: ref
  }) : null, React.createElement(Modal, {
    isOpen: isOpen,
    onClose: function onClose() {
      return onCloseModal();
    },
    showAttribution: showAttribution
  }, React.createElement(ModalContent, {
    appId: appId,
    consumerId: consumerId,
    jwt: jwt,
    onSelect: handleFileSelect,
    onConnectionSelect: handleConnectionSelect,
    title: title ? title : fileToSave ? 'Apideck File Uploader' : 'Apideck File Picker',
    subTitle: subTitle ? subTitle : fileToSave ? 'Select a folder' : 'Select a file',
    fileToSave: fileToSave
  })));
});

export { FilePicker };
